<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Introduction | HematoMap Tutorial</title>
  <meta name="description" content="This is the tutorial of HematoMap package." />
  <meta name="generator" content="bookdown 0.32 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Introduction | HematoMap Tutorial" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://nrctm-bioinfo.github.io/HematoMap/index.html/figure/cover.jpg" />
  <meta property="og:description" content="This is the tutorial of HematoMap package." />
  <meta name="github-repo" content="NRCTM-bioinfo/HematoMap" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Introduction | HematoMap Tutorial" />
  
  <meta name="twitter:description" content="This is the tutorial of HematoMap package." />
  <meta name="twitter:image" content="https://nrctm-bioinfo.github.io/HematoMap/index.html/figure/cover.jpg" />

<meta name="author" content="Yu-Ting Dai, Wen Ou-Yang" />


<meta name="date" content="2023-04-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="quick-start.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://github.com/NRCTM-bioinfo/HematoMap"><center><b>HematoMap</b></center></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Abstract</a></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#introduction-of-acute-leukemia"><i class="fa fa-check"></i><b>2.1</b> Introduction of acute leukemia</a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#what-can-hematomap-do"><i class="fa fa-check"></i><b>2.2</b> What can HematoMap do</a></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#data-resources-in-hematomap"><i class="fa fa-check"></i><b>2.3</b> Data resources in HematoMap</a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#outline-of-this-tutorial"><i class="fa fa-check"></i><b>2.4</b> Outline of this tutorial</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quick-start.html"><a href="quick-start.html"><i class="fa fa-check"></i><b>3</b> Quick-start</a>
<ul>
<li class="chapter" data-level="3.1" data-path="quick-start.html"><a href="quick-start.html#installation"><i class="fa fa-check"></i><b>3.1</b> Installation</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="quick-start.html"><a href="quick-start.html#installation-from-github"><i class="fa fa-check"></i><b>3.1.1</b> Installation from GitHub</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="quick-start.html"><a href="quick-start.html#load-package"><i class="fa fa-check"></i><b>3.2</b> Load package</a></li>
<li class="chapter" data-level="3.3" data-path="quick-start.html"><a href="quick-start.html#show-normal-bmmcs"><i class="fa fa-check"></i><b>3.3</b> Show normal BMMCs</a></li>
<li class="chapter" data-level="3.4" data-path="quick-start.html"><a href="quick-start.html#quick-start-code"><i class="fa fa-check"></i><b>3.4</b> Quick-start code</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="scrna.html"><a href="scrna.html"><i class="fa fa-check"></i><b>4</b> scRNA-seq</a>
<ul>
<li class="chapter" data-level="4.1" data-path="scrna.html"><a href="scrna.html#load-scrna-seq-data"><i class="fa fa-check"></i><b>4.1</b> Load scRNA-seq data</a></li>
<li class="chapter" data-level="4.2" data-path="scrna.html"><a href="scrna.html#calculate-similarities"><i class="fa fa-check"></i><b>4.2</b> Calculate similarities</a></li>
<li class="chapter" data-level="4.3" data-path="scrna.html"><a href="scrna.html#visualization"><i class="fa fa-check"></i><b>4.3</b> Visualization</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="bulk.html"><a href="bulk.html"><i class="fa fa-check"></i><b>5</b> Bulk RNA-seq</a>
<ul>
<li class="chapter" data-level="5.1" data-path="bulk.html"><a href="bulk.html#load-bulk-rna-seq-data"><i class="fa fa-check"></i><b>5.1</b> Load bulk RNA-seq data</a></li>
<li class="chapter" data-level="5.2" data-path="bulk.html"><a href="bulk.html#visualization-1"><i class="fa fa-check"></i><b>5.2</b> Visualization</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="about-hematomap.html"><a href="about-hematomap.html"><i class="fa fa-check"></i><b>6</b> About HematoMap</a>
<ul>
<li class="chapter" data-level="" data-path="about-hematomap.html"><a href="about-hematomap.html#bug-reports"><i class="fa fa-check"></i>Bug Reports</a></li>
<li class="chapter" data-level="" data-path="about-hematomap.html"><a href="about-hematomap.html#current-version"><i class="fa fa-check"></i>Current version</a></li>
<li class="chapter" data-level="" data-path="about-hematomap.html"><a href="about-hematomap.html#acknowledgment"><i class="fa fa-check"></i>Acknowledgment</a></li>
<li class="chapter" data-level="" data-path="about-hematomap.html"><a href="about-hematomap.html#session-information"><i class="fa fa-check"></i>Session Information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">HematoMap Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introduction" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Introduction<a href="introduction.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><img src="figure/logo.png" align="right" height=150 width=135/></p>
<div id="introduction-of-acute-leukemia" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Introduction of acute leukemia<a href="introduction.html#introduction-of-acute-leukemia" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Acute leukemia (AL) is a malignant disorder of hematopoiesis that arises from the clonal proliferation of leukemic blast cells. The presentation and progression of leukemia can vary greatly among different subgroups of patients. Additionally, within the same patient, leukemia cells with different characteristics can coexist. There are two distinct models for understanding cellular heterogeneity and cell-of-origin of acute leukemia: one is the genomic model, and the other is the stem-cell model. The genomic model of acute leukemia is based on the identification of genomic lesions, such as recurrent somatic mutations or fusions. These mutations/fusions are involved in several categories, including transcriptional regulation, signaling pathways, and epigenetic modifiers. However, although the genomic model can explain much of the heterogeneity in leukemia, even cells with the same mutation may have different functions, so the explanatory power of the genomic model is limited.</p>
<p>Another classic model is the hematopoietic stem cell (HSC) model, which suggests that leukemia is caused by a small population of self-renewing and differentiation-arrested stem cells (also known as leukemia stem cells, LSC). These leukemia stem cells possess characteristics similar to normal hematopoietic stem cells, such as the ability to self-renew and produce different types of other blood cells, while other leukemia cells are their differentiated progeny. According to this model, leukemia is a hierarchical disease in which leukemia stem cells are located at the top of the hierarchy, while more differentiated leukemia cells are at the bottom. The hematopoietic stem cell model provides a better explanation for leukemia heterogeneity, as it can explain why different types of leukemia cells can exist even within the same patient.</p>
<p>The differentiation trajectory of hematopoiesis in a healthy human body is extremely complex and crucial for understanding the mechanism of leukemia. In the past, methods such as bone marrow tissue sectioning and cell culture were commonly used to understand the differentiation process of hematopoietic cells. However, these methods still have limitations, such as the inability to comprehensively and accurately reflect the phenotypic and functional characteristics of hematopoietic cells at different developmental stages. In recent years, with the development of sequencing technologies, especially single-cell sequencing, new insights has been brought into the hematopoietic cell differentiation exploration.</p>
</div>
<div id="what-can-hematomap-do" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> What can HematoMap do<a href="introduction.html#what-can-hematomap-do" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Here we present <strong>HematoMap</strong>, a comprehensive R package that can provide a visual representation of the cellular hematopoietic hierarchy of human bone marrow. It is a mapping tool that integrates information from single-cell based gene expression data and generate a unified model of hematopoietic cell differentiation. HematoMap allows researchers to explore the relationships between different hematopoietic cell types (such as HSC/MPP, CML and CLP) and interpret normal hematopoiesis cellular compositions. In this <strong>HematoMap</strong> project, by re-collecting, re-analyzing, and re-annotating single-cell RNA sequencing (scRNA-seq) data of normal bone marrow mononuclear cells (BMMCs), we constructed a hierarchy-based reference to interpret normal hematopoiesis cellular compositions. A likelihood score was also developed to comprehensively and accurately project leukemic blast cells onto the cellular hierarchy of normal hematopoietic differentiation. The identification of the cell-of-origin and resolution of heterogeneity of acute leukemia is an important long-term goal, as it may provide critical insights into the pathogenesis of this disease and inform the development of novel therapeutic strategies.</p>
<p>Using HematoMap, you can do:</p>
<ul>
<li><p><strong>1</strong> Visual representation of the cellular composition of normal human BMMCs.</p></li>
<li><p><strong>2</strong> Visual representation of cell blockage of acute leukemia.</p></li>
<li><p><strong>3</strong> Lineage aberrancy inference using scRNA-seq of acute leukemia.</p></li>
<li><p><strong>4</strong> Lineage aberrancy inference using bulk RNA-seq of acute leukemia.</p></li>
</ul>
</div>
<div id="data-resources-in-hematomap" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Data resources in HematoMap<a href="introduction.html#data-resources-in-hematomap" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The human hematopoietic hierarchy in bone marrow consists of a hierarchical system of cells that differentiate from hematopoietic stem cells (HSCs) to mature blood cells. The hierarchy includes many kinds of progenitors such as multipotent progenitor cells (MPPs), common lymphoid progenitor cells (CLPs), common myeloid progenitor cells (CMPs), megakaryocyte-erythroid progenitor cells (MEPs), and granulocyte-macrophage progenitor cells (GMPs). These progenitor cells differentiate further into more specialized cells, such as erythrocytes, megakaryocytes, granulocytes, monocytes, and lymphocytes. In <strong>HematoMap</strong>, we provided detailed annotation of hematopoietic cell differentiation based on single-cell data at different hierarchy levels.</p>
<p>The annotation of healthy BMMCs could be described as four levels:</p>
<ul>
<li><p><strong>Level 1: Lineages</strong>. HSPCs, Myeloids, Bcells, T/NKcells.</p></li>
<li><p><strong>Level 2: Populations</strong>. HSPCs, Monocytes, Dendritic cells, Erythrocytes, pro B cells, mature B cells, CD8+ T cells, CD4+ T cells, NK cells</p></li>
<li><p><strong>Level 3: Types</strong>. The 42 cell types (including 4 cell cycle related cell types): HSC/MPP, LMPP, CLP, CMP, MDP, GMP, CDP, MEP, pro-Mono, CD14 Mono, CD16 Mono, pre-DC, pDC, cDC1, mo-DC, MKP, MK, pro-Ery1, pro-Ery2, Ery, pre-pro-B, Early pro-B, 23 Early cycling pro-B, Late pro-B, Late cycling pro-B, pre-B, Immature B, Naive B, Memory B 1, Memory B 2, CD8 Tnaive, CD8 Teff, CD8 Tex, CD8 Tdpe, CD8 Tmpe, CD4 Tnaive, CD4 Tem, CD4 Treg, NK, NK-XCL1, cycling NK, cycling NK/T.</p></li>
<li><p><strong>Level 4: Sub-clusters</strong>. The 1079 cell sub-clusters, for example, HSC/MPP:0001, HSC/MPP:0002, HSC/MPP:0003.</p></li>
</ul>
<p>The relationship of the four levels and the related marker for cell type identification were listed as followed.</p>
<table>
<caption><span id="tab:nice-tab">Table 2.1: </span>Structure of hierarchy-based cell types in normal BMMCs</caption>
<colgroup>
<col width="4%" />
<col width="7%" />
<col width="9%" />
<col width="21%" />
<col width="39%" />
<col width="18%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Lineage</th>
<th align="left">Population</th>
<th align="left">CellType</th>
<th align="left">FullName</th>
<th align="left">Markers</th>
<th align="left">Reference..PMID.</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HSPCs</td>
<td align="left">HSPCs</td>
<td align="left">HSC/MPP</td>
<td align="left">Hematopoietic stem cell/Multipotent progenitor</td>
<td align="left">CD34(+), KIT(+), FLT3(-), AVP(+), CD38(-),</td>
<td align="left">28650480;</td>
</tr>
<tr class="even">
<td align="left">HSPCs</td>
<td align="left">HSPCs</td>
<td align="left">LMPP</td>
<td align="left">Lymphoid-primed multipotent progenitor</td>
<td align="left">CD34(+), KIT(+), FLT3(+), AVP(+), CD38(-),</td>
<td align="left">28650480;</td>
</tr>
<tr class="odd">
<td align="left">HSPCs</td>
<td align="left">HSPCs</td>
<td align="left">CLP</td>
<td align="left">Common lymphoid progenitor</td>
<td align="left">CD34(+), KIT(+), FLT3(+), MME(+/-), CD7(+/-),</td>
<td align="left">28650480; 19996414;</td>
</tr>
<tr class="even">
<td align="left">HSPCs</td>
<td align="left">HSPCs</td>
<td align="left">CMP</td>
<td align="left">Common myeloid progenitor</td>
<td align="left">CD34(+), KIT(+), FLT3(+), CD38(+), CSF1R(-), FCGR1A(-),</td>
<td align="left">28650480;</td>
</tr>
<tr class="odd">
<td align="left">HSPCs</td>
<td align="left">HSPCs</td>
<td align="left">MDP</td>
<td align="left">Monocyte-DC progenitor</td>
<td align="left">CD34(+), KIT(+), FLT3(+), CD38(+), CSF1R(+), FCGR1A(+/-),</td>
<td align="left">29166589; 28650480; 28514689; 29313948;</td>
</tr>
<tr class="even">
<td align="left">HSPCs</td>
<td align="left">HSPCs</td>
<td align="left">GMP</td>
<td align="left">Granulocyte-macrophage progenitor</td>
<td align="left">CD34(+), KIT(+), FLT3(+), CD38(+), MPO(+), ELANE(+), FCGR1A(+),</td>
<td align="left">28650480; 26332381;</td>
</tr>
<tr class="odd">
<td align="left">HSPCs</td>
<td align="left">HSPCs</td>
<td align="left">CDP</td>
<td align="left">Common DC progenitor</td>
<td align="left">CD34(+), KIT(+), FLT3(+), CD38(+), CSF1R(+), FCGR1A(-), IL3RA(+),</td>
<td align="left">28650480; 29313948;</td>
</tr>
<tr class="even">
<td align="left">HSPCs</td>
<td align="left">HSPCs</td>
<td align="left">MEP</td>
<td align="left">Megakaryocyte-erythrocyte progenitor</td>
<td align="left">CD34(+), KIT(+), FLT3(-), CD38(+),</td>
<td align="left">28650480;</td>
</tr>
<tr class="odd">
<td align="left">Myeloids</td>
<td align="left">Monocytes</td>
<td align="left">pro-Mono</td>
<td align="left">Promonocytes</td>
<td align="left">CD34(-), KIT(-), CD300E(-), LYZ(+), CD36(+), MPO(+), FCGR1A(+), CD4(+)</td>
<td align="left">26282340; 28514689;</td>
</tr>
<tr class="even">
<td align="left">Myeloids</td>
<td align="left">Monocytes</td>
<td align="left">CD14 Mono</td>
<td align="left">Classical(CD14+CD16−) monocyte</td>
<td align="left">CD14(+), FCGR3A(-), CD300E(+), LYZ(+), ITGAX(+), FCGR1A(+), CD4(+)</td>
<td align="left">26332381; 28514689;</td>
</tr>
<tr class="odd">
<td align="left">Myeloids</td>
<td align="left">Monocytes</td>
<td align="left">CD16 Mono</td>
<td align="left">Non-classical(CD14−CD16+) monocyte</td>
<td align="left">CD14(-), FCGR3A(+), CD300E(+), LYZ(+), ITGAX(+), CD4(+)</td>
<td align="left">26332381; 28514689;</td>
</tr>
<tr class="even">
<td align="left">Myeloids</td>
<td align="left">Dendritic cells</td>
<td align="left">pre-DC</td>
<td align="left">Dendritic cell precursor</td>
<td align="left">CD34(-), FLT3(+), AXL(+), SIGLEC6(+), ITGAX(+),</td>
<td align="left">29313948;</td>
</tr>
<tr class="odd">
<td align="left">Myeloids</td>
<td align="left">Dendritic cells</td>
<td align="left">pDC</td>
<td align="left">Plasmacytoid dendritic cell</td>
<td align="left">CLEC4C(+), IRF4(+), LILRA4(+), IL3RA(+), IRF8(+), IRF7(+),</td>
<td align="left">29313948; 20193018;</td>
</tr>
<tr class="even">
<td align="left">Myeloids</td>
<td align="left">Dendritic cells</td>
<td align="left">cDC1</td>
<td align="left">Conventional dendritic cell 1</td>
<td align="left">XCR1(+), CD1C(+), ITGAX(+), IRF8(+), THBD(+),</td>
<td align="left">29313948;</td>
</tr>
<tr class="odd">
<td align="left">Myeloids</td>
<td align="left">Dendritic cells</td>
<td align="left">mo-DC</td>
<td align="left">Monocyte-derived dendritic cell</td>
<td align="left">MRC1(+), LYZ(+), ITGAX(+),</td>
<td align="left">29313948;</td>
</tr>
<tr class="even">
<td align="left">Myeloids</td>
<td align="left">Erythrocytes</td>
<td align="left">MKP</td>
<td align="left">Megakaryocyte progenitor</td>
<td align="left">CD34(+), KIT(+), IFLT3(-), ITGA2B(+),</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">Myeloids</td>
<td align="left">Erythrocytes</td>
<td align="left">MK</td>
<td align="left">Megakaryocyte</td>
<td align="left">PF4(+), ITGA2B(+), CD9(+),</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">Myeloids</td>
<td align="left">Erythrocytes</td>
<td align="left">pro-Ery1</td>
<td align="left">Proerythroblast 1</td>
<td align="left">ENG(+), KLF(+), KIT(+), TFRC(+)</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">Myeloids</td>
<td align="left">Erythrocytes</td>
<td align="left">pro-Ery2</td>
<td align="left">Proerythroblast 2</td>
<td align="left">ENG(+), KLF(+), KIT(-), TFRC(+)</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">Myeloids</td>
<td align="left">Erythrocytes</td>
<td align="left">Ery</td>
<td align="left">Erythroblast</td>
<td align="left">ENG(-), KLF(+), TFRC(+)</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">B cells</td>
<td align="left">B cells</td>
<td align="left">pre-pro-B</td>
<td align="left">B cell progenitor precursor</td>
<td align="left">CD79A(+), IGLL1(+), RAG1(+), RAG2(+), VPREB1(+), MME(+), IL7R(+),</td>
<td align="left">35090788; 19996414;</td>
</tr>
<tr class="even">
<td align="left">B cells</td>
<td align="left">B cells</td>
<td align="left">Early pro-B</td>
<td align="left">Early B cell progenitor</td>
<td align="left">CD79A(+), DNTT(+), IGLL1(+), RAG1(+), RAG2(+), VPREB1(+),</td>
<td align="left">35090788;</td>
</tr>
<tr class="odd">
<td align="left">B cells</td>
<td align="left">B cells</td>
<td align="left">Early cycling pro-B</td>
<td align="left">Early cycling B cell progenitor</td>
<td align="left">MKI67(+), PCNA(+), CD79A(+), DNTT(+), IGLL1(+), RAG1(+), RAG2(+), VPREB1(+), IL7R(-),</td>
<td align="left">35090788;</td>
</tr>
<tr class="even">
<td align="left">B cells</td>
<td align="left">B cells</td>
<td align="left">Late pro-B</td>
<td align="left">Late B cell progenitor</td>
<td align="left">CD79A(+), DNTT(-), IGLL1(+), TCL1A(+), RAG1(+), RAG2(+), VPREB1(+),</td>
<td align="left">35090788;</td>
</tr>
<tr class="odd">
<td align="left">B cells</td>
<td align="left">B cells</td>
<td align="left">Late cycling pro-B</td>
<td align="left">Late cycling B cell progenitor</td>
<td align="left">MKI67(+), PCNA(+), CD79A(+), DNTT(-), IGLL1(+), TCL1A(+), RAG1(+), RAG2(+), VPREB1(+),</td>
<td align="left">35090788;</td>
</tr>
<tr class="even">
<td align="left">B cells</td>
<td align="left">B cells</td>
<td align="left">pre-B</td>
<td align="left">B cell precursor</td>
<td align="left">IGLL1(+/-), TCL1A(+), RAG1(+), RAG2(+), VPREB1(+),</td>
<td align="left">35090788;</td>
</tr>
<tr class="odd">
<td align="left">B cells</td>
<td align="left">B cells</td>
<td align="left">Immature B</td>
<td align="left">Immature B cell</td>
<td align="left">MS4A1(+), TCL1A(+), RAG1(-), RAG2(-), VPREB1(+/-),</td>
<td align="left">35090788;</td>
</tr>
<tr class="even">
<td align="left">B cells</td>
<td align="left">B cells</td>
<td align="left">Naive B</td>
<td align="left">Naive B cell</td>
<td align="left">MS4A1(+), CD27(-), IGHD(+), TCL1A(+),</td>
<td align="left">35090788;</td>
</tr>
<tr class="odd">
<td align="left">B cells</td>
<td align="left">B cells</td>
<td align="left">Memory B 1</td>
<td align="left">Memory B cell 1</td>
<td align="left">MS4A1(+), CD27(+), IGHD(+), IGHG3(+/-),</td>
<td align="left">35090788;</td>
</tr>
<tr class="even">
<td align="left">B cells</td>
<td align="left">B cells</td>
<td align="left">Memory B 2</td>
<td align="left">Memory B cell 2</td>
<td align="left">MS4A1(-), CD27(+), IGHD(-), IGHG3(++),</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">T/NK cells</td>
<td align="left">CD8+ T cells</td>
<td align="left">CD8 Tnaive</td>
<td align="left">Naive CD8+ T cell</td>
<td align="left">CD3D(+), CD3E(+), CD8A(+), CCR7(+), IL7R(+), SELL(+)</td>
<td align="left">29630679;</td>
</tr>
<tr class="even">
<td align="left">T/NK cells</td>
<td align="left">CD8+ T cells</td>
<td align="left">CD8 Tdpe</td>
<td align="left">KLRG1+IL7R+ double-positive effector CD8+ T</td>
<td align="left">CD3D(+), CD3E(+), CD8A(+), IL7R(+), CCR7(-), KLRG1(+), CD27(+)</td>
<td align="left">29625895;</td>
</tr>
<tr class="odd">
<td align="left">T/NK cells</td>
<td align="left">CD8+ T cells</td>
<td align="left">CD8 Tmpe</td>
<td align="left">Memory precursor effector CD8+ T</td>
<td align="left">CD3D(+), CD3E(+), CD8A(+), IL7R(+), CCR7(-), KLRG1(dim/-)</td>
<td align="left">29625895;</td>
</tr>
<tr class="even">
<td align="left">T/NK cells</td>
<td align="left">CD8+ T cells</td>
<td align="left">CD8 Teff</td>
<td align="left">Effector CD8+ T cell</td>
<td align="left">CD3D(+), CD3E(+), CD8A(+), CCR7(-), GNLY(+), NKG7(+)</td>
<td align="left">34017005;</td>
</tr>
<tr class="odd">
<td align="left">T/NK cells</td>
<td align="left">CD8+ T cells</td>
<td align="left">CD8 Tex</td>
<td align="left">Exhausted CD8+ T</td>
<td align="left">CD3D(+), CD3E(+), CD8A(+), CCR7(-), PDCD1(+), TNFRSF9(+), LAG3(+), CD160(+),</td>
<td align="left">34017005;</td>
</tr>
<tr class="even">
<td align="left">T/NK cells</td>
<td align="left">CD4+ T cells</td>
<td align="left">CD4 Tnaive</td>
<td align="left">Naive CD4+ T cell</td>
<td align="left">CD3D(+), CD3E(+), CD4(+), CCR7(+), IL7R(+), SELL(+), CD40LG(+),</td>
<td align="left">29630679;</td>
</tr>
<tr class="odd">
<td align="left">T/NK cells</td>
<td align="left">CD4+ T cells</td>
<td align="left">CD4 Tem</td>
<td align="left">Memory CD4+ T cell</td>
<td align="left">CD3D(+), CD3E(+), CD4(+), CCR7(-), IL7R(+), CD40LG(+),</td>
<td align="left">29630679;</td>
</tr>
<tr class="even">
<td align="left">T/NK cells</td>
<td align="left">CD4+ T cells</td>
<td align="left">CD4 Treg</td>
<td align="left">Regulatory T cell</td>
<td align="left">CD3D(+), CD3E(+), CD4(+), IL2RA(+), FOXP3(+), DUSP4(+)</td>
<td align="left">34017005;</td>
</tr>
<tr class="odd">
<td align="left">T/NK cells</td>
<td align="left">NK</td>
<td align="left">NK</td>
<td align="left">Natural killer cell</td>
<td align="left">IL2RB(+), IL7R(-), KLRF1(+), XCL1(-), FCGR3A(+), NCAM1(+),</td>
<td align="left">30150991;</td>
</tr>
<tr class="even">
<td align="left">T/NK cells</td>
<td align="left">NK</td>
<td align="left">NK-XCL1</td>
<td align="left">XCL1+ Natural killer cell</td>
<td align="left">IL2RB(+), IL7R(-), KLRF1(+), XCL1(+), NCAM1(+),</td>
<td align="left">30150991;</td>
</tr>
<tr class="odd">
<td align="left">T/NK cells</td>
<td align="left">NK</td>
<td align="left">cycling NK</td>
<td align="left">Cycling natural killer cell</td>
<td align="left">MKI67(+), PCNA(+), IL2RB(+), IL7R(-), KLRF1(+), FCGR3A(+), NCAM1(+),</td>
<td align="left">30150991;</td>
</tr>
<tr class="even">
<td align="left">T/NK cells</td>
<td align="left">NK</td>
<td align="left">cycling NK/T</td>
<td align="left">Cycling natural killer/T cell</td>
<td align="left">MKI67(+), PCNA(+), CD3D(+), CD3E(+), KLRF(+), IL2RB(+/-), NCAM1(+)</td>
<td align="left">30150991;</td>
</tr>
</tbody>
</table>
</div>
<div id="outline-of-this-tutorial" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Outline of this tutorial<a href="introduction.html#outline-of-this-tutorial" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <a href="https://github.com/NRCTM-bioinfo/HematoMap"><strong><code>HematoMap</code></strong></a> is an R-based package for lineage aberrancy inference based on single-cell RNA sequencing (<strong>scRNA-seq</strong>) and bulk RNA sequencing (<strong>bulk RNA-seq</strong>) data of acute leukemia, including . In this tutorial, we will present:</p>
<ul>
<li><ol style="list-style-type: decimal">
<li>How to install <code>HematoMap</code>.</li>
</ol></li>
<li><ol start="2" style="list-style-type: decimal">
<li>Overview of workflow of <code>HematoMap</code>.</li>
</ol></li>
<li><ol start="3" style="list-style-type: decimal">
<li>scRNA-seq analysis using <code>HematoMap</code>.</li>
</ol></li>
<li><ol start="4" style="list-style-type: decimal">
<li>Bulk RNA-seq analysis using <code>HematoMap</code>.</li>
</ol></li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="quick-start.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/01-Introduction.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["HematoMap-tutorial.pdf", "HematoMap-tutorial.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
