<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 About this project | p53 compounds</title>
  <meta name="description" content="This is the tutorial of p53 compounds project." />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 About this project | p53 compounds" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the tutorial of p53 compounds project." />
  <meta name="github-repo" content="NRCTM-bioinfo/p53-compounds" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 About this project | p53 compounds" />
  
  <meta name="twitter:description" content="This is the tutorial of p53 compounds project." />
  

<meta name="author" content="Yu-Ting Dai" />


<meta name="date" content="2024-01-13" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="tcga-pancanatlas.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="http://www.rescuep53.net/">p53 compounds</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="data-preprocessing.html"><a href="data-preprocessing.html"><i class="fa fa-check"></i><b>2</b> Data Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#rna-seq-data"><i class="fa fa-check"></i><b>2.1</b> RNA-seq data</a></li>
<li class="chapter" data-level="2.2" data-path="data-preprocessing.html"><a href="data-preprocessing.html#geo-data"><i class="fa fa-check"></i><b>2.2</b> GEO data</a></li>
<li class="chapter" data-level="2.3" data-path="data-preprocessing.html"><a href="data-preprocessing.html#source-data"><i class="fa fa-check"></i><b>2.3</b> Source data</a></li>
<li class="chapter" data-level="2.4" data-path="data-preprocessing.html"><a href="data-preprocessing.html#tcga-data"><i class="fa fa-check"></i><b>2.4</b> TCGA data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="target-evaluation.html"><a href="target-evaluation.html"><i class="fa fa-check"></i><b>3</b> Target Evaluation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="target-evaluation.html"><a href="target-evaluation.html#load-packages"><i class="fa fa-check"></i><b>3.1</b> Load packages</a></li>
<li class="chapter" data-level="3.2" data-path="target-evaluation.html"><a href="target-evaluation.html#merge-profiling"><i class="fa fa-check"></i><b>3.2</b> Merge profiling</a></li>
<li class="chapter" data-level="3.3" data-path="target-evaluation.html"><a href="target-evaluation.html#normalization"><i class="fa fa-check"></i><b>3.3</b> Normalization</a></li>
<li class="chapter" data-level="3.4" data-path="target-evaluation.html"><a href="target-evaluation.html#confident-p53-targets"><i class="fa fa-check"></i><b>3.4</b> 32 confident p53 targets</a></li>
<li class="chapter" data-level="3.5" data-path="target-evaluation.html"><a href="target-evaluation.html#p53-targets"><i class="fa fa-check"></i><b>3.5</b> 116 p53 targets</a></li>
<li class="chapter" data-level="3.6" data-path="target-evaluation.html"><a href="target-evaluation.html#p53-transactivation"><i class="fa fa-check"></i><b>3.6</b> p53 transactivation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html"><i class="fa fa-check"></i><b>4</b> Enrichment analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#calculated-degs"><i class="fa fa-check"></i><b>4.1</b> Calculated DEGs</a></li>
<li class="chapter" data-level="4.2" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#enrichr-analysis"><i class="fa fa-check"></i><b>4.2</b> Enrichr analysis</a></li>
<li class="chapter" data-level="4.3" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#msigdb-hallmark"><i class="fa fa-check"></i><b>4.3</b> MSigDB Hallmark</a></li>
<li class="chapter" data-level="4.4" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#kegg"><i class="fa fa-check"></i><b>4.4</b> KEGG</a></li>
<li class="chapter" data-level="4.5" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#trrust"><i class="fa fa-check"></i><b>4.5</b> TRRUST</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="cmap-database.html"><a href="cmap-database.html"><i class="fa fa-check"></i><b>5</b> CMAP database</a></li>
<li class="chapter" data-level="6" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html"><i class="fa fa-check"></i><b>6</b> TCGA PanCanAtlas</a>
<ul>
<li class="chapter" data-level="6.1" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#wt-up-genes"><i class="fa fa-check"></i><b>6.1</b> WT-up genes</a></li>
<li class="chapter" data-level="6.2" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#mut-up-genes"><i class="fa fa-check"></i><b>6.2</b> MUT-up genes</a></li>
<li class="chapter" data-level="6.3" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#hr-of-high-expression"><i class="fa fa-check"></i><b>6.3</b> HR of high expression</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#p53-targets-1"><i class="fa fa-check"></i><b>6.3.1</b> 116 p53 targets</a></li>
<li class="chapter" data-level="6.3.2" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#wt-up-genes-1"><i class="fa fa-check"></i><b>6.3.2</b> WT-up genes</a></li>
<li class="chapter" data-level="6.3.3" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#mut-up-genes-1"><i class="fa fa-check"></i><b>6.3.3</b> MUT-up genes</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#powerfulness-in-regulating"><i class="fa fa-check"></i><b>6.4</b> Powerfulness in regulating</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#p53-targets-2"><i class="fa fa-check"></i><b>6.4.1</b> 116 p53 targets</a></li>
<li class="chapter" data-level="6.4.2" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#wt-up-genes-2"><i class="fa fa-check"></i><b>6.4.2</b> WT-up genes</a></li>
<li class="chapter" data-level="6.4.3" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#mut-up-genes-2"><i class="fa fa-check"></i><b>6.4.3</b> MUT-up genes</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#hr-for-p53-mutations"><i class="fa fa-check"></i><b>6.5</b> HR for p53 mutations</a></li>
<li class="chapter" data-level="6.6" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#mdm2-expression"><i class="fa fa-check"></i><b>6.6</b> MDM2 expression</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#radar-plot"><i class="fa fa-check"></i><b>6.6.1</b> Radar plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="about-this-project.html"><a href="about-this-project.html"><i class="fa fa-check"></i><b>7</b> About this project</a>
<ul>
<li class="chapter" data-level="" data-path="about-this-project.html"><a href="about-this-project.html#current-version"><i class="fa fa-check"></i>Current version</a></li>
<li class="chapter" data-level="" data-path="about-this-project.html"><a href="about-this-project.html#session-information"><i class="fa fa-check"></i>Session Information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">p53 compounds</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="about-this-project" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> About this project<a href="about-this-project.html#about-this-project" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="current-version" class="section level2 unnumbered hasAnchor">
<h2>Current version<a href="about-this-project.html#current-version" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Version: 1.2.1</p>
<p>Release Date: 2024-01-13</p>
</div>
<div id="session-information" class="section level2 unnumbered hasAnchor">
<h2>Session Information<a href="about-this-project.html#session-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="about-this-project.html#cb61-1" tabindex="-1"></a><span class="co"># Show session information</span></span>
<span id="cb61-2"><a href="about-this-project.html#cb61-2" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.3.2 (2023-10-31)
## Platform: x86_64-apple-darwin20 (64-bit)
## Running under: macOS Ventura 13.0.1
## 
## Matrix products: default
## BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## time zone: Asia/Shanghai
## tzcode source: internal
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] fmsb_0.7.5                  ggthemes_5.0.0              stringr_1.5.1              
##  [4] enrichR_3.2                 survival_3.5-7              survminer_0.4.9            
##  [7] gplots_3.1.3                pheatmap_1.0.12             limma_3.58.1               
## [10] ggpubr_0.6.0                ggplot2_3.4.4               data.table_1.14.10         
## [13] dendextend_1.17.1           DESeq2_1.42.0               SummarizedExperiment_1.32.0
## [16] Biobase_2.62.0              MatrixGenerics_1.14.0       matrixStats_1.2.0          
## [19] GenomicRanges_1.54.1        GenomeInfoDb_1.38.1         IRanges_2.36.0             
## [22] S4Vectors_0.40.2            BiocGenerics_0.48.1         nlme_3.1-164               
## [25] DT_0.31                     dplyr_1.1.4                 openxlsx_4.2.5.2           
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-7            gridExtra_2.3           rlang_1.1.2             magrittr_2.0.3         
##  [5] compiler_4.3.2          vctrs_0.6.5             pkgconfig_2.0.3         crayon_1.5.2           
##  [9] fastmap_1.1.1           backports_1.4.1         XVector_0.42.0          labeling_0.4.3         
## [13] KMsurv_0.1-5            caTools_1.18.2          utf8_1.2.4              rmarkdown_2.25         
## [17] purrr_1.0.2             xfun_0.41               WriteXLS_6.4.0          zlibbioc_1.48.0        
## [21] cachem_1.0.8            jsonlite_1.8.8          highr_0.10              DelayedArray_0.28.0    
## [25] BiocParallel_1.36.0     broom_1.0.5             parallel_4.3.2          R6_2.5.1               
## [29] bslib_0.6.1             stringi_1.8.3           RColorBrewer_1.1-3      car_3.1-2              
## [33] jquerylib_0.1.4         Rcpp_1.0.11             bookdown_0.37           knitr_1.45             
## [37] zoo_1.8-12              Matrix_1.6-4            splines_4.3.2           tidyselect_1.2.0       
## [41] rstudioapi_0.15.0       abind_1.4-5             yaml_2.3.8              viridis_0.6.4          
## [45] codetools_0.2-19        curl_5.2.0              lattice_0.22-5          tibble_3.2.1           
## [49] withr_2.5.2             evaluate_0.23           zip_2.3.0               survMisc_0.5.6         
## [53] pillar_1.9.0            carData_3.0-5           KernSmooth_2.23-22      generics_0.1.3         
## [57] RCurl_1.98-1.13         munsell_0.5.0           scales_1.3.0            gtools_3.9.5           
## [61] xtable_1.8-4            glue_1.6.2              tools_4.3.2             locfit_1.5-9.8         
## [65] ggsignif_0.6.4          grid_4.3.2              tidyr_1.3.0             colorspace_2.1-0       
## [69] GenomeInfoDbData_1.2.11 cli_3.6.2               km.ci_0.5-6             fansi_1.0.6            
## [73] S4Arrays_1.2.0          viridisLite_0.4.2       gtable_0.3.4            rstatix_0.7.2          
## [77] sass_0.4.8              digest_0.6.33           SparseArray_1.2.2       farver_2.1.1           
## [81] rjson_0.2.21            htmlwidgets_1.6.4       htmltools_0.5.7         lifecycle_1.0.4        
## [85] httr_1.4.7              statmod_1.5.0</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="tcga-pancanatlas.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/06-About.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["p53-compounds.pdf", "p53-compounds.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
