<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Data Preprocessing | p53 compounds</title>
  <meta name="description" content="This is the tutorial of p53 compounds project." />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Data Preprocessing | p53 compounds" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the tutorial of p53 compounds project." />
  <meta name="github-repo" content="NRCTM-bioinfo/p53-compounds" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Data Preprocessing | p53 compounds" />
  
  <meta name="twitter:description" content="This is the tutorial of p53 compounds project." />
  

<meta name="author" content="Yu-Ting Dai" />


<meta name="date" content="2024-01-13" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="target-evaluation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="http://www.rescuep53.net/">p53 compounds</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="data-preprocessing.html"><a href="data-preprocessing.html"><i class="fa fa-check"></i><b>2</b> Data Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-preprocessing.html"><a href="data-preprocessing.html#rna-seq-data"><i class="fa fa-check"></i><b>2.1</b> RNA-seq data</a></li>
<li class="chapter" data-level="2.2" data-path="data-preprocessing.html"><a href="data-preprocessing.html#geo-data"><i class="fa fa-check"></i><b>2.2</b> GEO data</a></li>
<li class="chapter" data-level="2.3" data-path="data-preprocessing.html"><a href="data-preprocessing.html#source-data"><i class="fa fa-check"></i><b>2.3</b> Source data</a></li>
<li class="chapter" data-level="2.4" data-path="data-preprocessing.html"><a href="data-preprocessing.html#tcga-data"><i class="fa fa-check"></i><b>2.4</b> TCGA data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="target-evaluation.html"><a href="target-evaluation.html"><i class="fa fa-check"></i><b>3</b> Target Evaluation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="target-evaluation.html"><a href="target-evaluation.html#load-packages"><i class="fa fa-check"></i><b>3.1</b> Load packages</a></li>
<li class="chapter" data-level="3.2" data-path="target-evaluation.html"><a href="target-evaluation.html#merge-profiling"><i class="fa fa-check"></i><b>3.2</b> Merge profiling</a></li>
<li class="chapter" data-level="3.3" data-path="target-evaluation.html"><a href="target-evaluation.html#normalization"><i class="fa fa-check"></i><b>3.3</b> Normalization</a></li>
<li class="chapter" data-level="3.4" data-path="target-evaluation.html"><a href="target-evaluation.html#confident-p53-targets"><i class="fa fa-check"></i><b>3.4</b> 32 confident p53 targets</a></li>
<li class="chapter" data-level="3.5" data-path="target-evaluation.html"><a href="target-evaluation.html#p53-targets"><i class="fa fa-check"></i><b>3.5</b> 116 p53 targets</a></li>
<li class="chapter" data-level="3.6" data-path="target-evaluation.html"><a href="target-evaluation.html#p53-transactivation"><i class="fa fa-check"></i><b>3.6</b> p53 transactivation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html"><i class="fa fa-check"></i><b>4</b> Enrichment analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#calculated-degs"><i class="fa fa-check"></i><b>4.1</b> Calculated DEGs</a></li>
<li class="chapter" data-level="4.2" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#enrichr-analysis"><i class="fa fa-check"></i><b>4.2</b> Enrichr analysis</a></li>
<li class="chapter" data-level="4.3" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#msigdb-hallmark"><i class="fa fa-check"></i><b>4.3</b> MSigDB Hallmark</a></li>
<li class="chapter" data-level="4.4" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#kegg"><i class="fa fa-check"></i><b>4.4</b> KEGG</a></li>
<li class="chapter" data-level="4.5" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#trrust"><i class="fa fa-check"></i><b>4.5</b> TRRUST</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="cmap-database.html"><a href="cmap-database.html"><i class="fa fa-check"></i><b>5</b> CMAP database</a></li>
<li class="chapter" data-level="6" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html"><i class="fa fa-check"></i><b>6</b> TCGA PanCanAtlas</a>
<ul>
<li class="chapter" data-level="6.1" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#wt-up-genes"><i class="fa fa-check"></i><b>6.1</b> WT-up genes</a></li>
<li class="chapter" data-level="6.2" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#mut-up-genes"><i class="fa fa-check"></i><b>6.2</b> MUT-up genes</a></li>
<li class="chapter" data-level="6.3" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#hr-of-high-expression"><i class="fa fa-check"></i><b>6.3</b> HR of high expression</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#p53-targets-1"><i class="fa fa-check"></i><b>6.3.1</b> 116 p53 targets</a></li>
<li class="chapter" data-level="6.3.2" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#wt-up-genes-1"><i class="fa fa-check"></i><b>6.3.2</b> WT-up genes</a></li>
<li class="chapter" data-level="6.3.3" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#mut-up-genes-1"><i class="fa fa-check"></i><b>6.3.3</b> MUT-up genes</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#powerfulness-in-regulating"><i class="fa fa-check"></i><b>6.4</b> Powerfulness in regulating</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#p53-targets-2"><i class="fa fa-check"></i><b>6.4.1</b> 116 p53 targets</a></li>
<li class="chapter" data-level="6.4.2" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#wt-up-genes-2"><i class="fa fa-check"></i><b>6.4.2</b> WT-up genes</a></li>
<li class="chapter" data-level="6.4.3" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#mut-up-genes-2"><i class="fa fa-check"></i><b>6.4.3</b> MUT-up genes</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#hr-for-p53-mutations"><i class="fa fa-check"></i><b>6.5</b> HR for p53 mutations</a></li>
<li class="chapter" data-level="6.6" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#mdm2-expression"><i class="fa fa-check"></i><b>6.6</b> MDM2 expression</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="tcga-pancanatlas.html"><a href="tcga-pancanatlas.html#radar-plot"><i class="fa fa-check"></i><b>6.6.1</b> Radar plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="about-this-project.html"><a href="about-this-project.html"><i class="fa fa-check"></i><b>7</b> About this project</a>
<ul>
<li class="chapter" data-level="" data-path="about-this-project.html"><a href="about-this-project.html#current-version"><i class="fa fa-check"></i>Current version</a></li>
<li class="chapter" data-level="" data-path="about-this-project.html"><a href="about-this-project.html#session-information"><i class="fa fa-check"></i>Session Information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">p53 compounds</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-preprocessing" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Data Preprocessing<a href="data-preprocessing.html#data-preprocessing" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="rna-seq-data" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> RNA-seq data<a href="data-preprocessing.html#rna-seq-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The expression matrix of ATO and PAT was generated from raw RNA-sequencing data (FASTQ format) using salmon (<a href="https://salmon.readthedocs.io/en/latest/salmon.html" class="uri">https://salmon.readthedocs.io/en/latest/salmon.html</a>).</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="data-preprocessing.html#cb1-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb1-2"><a href="data-preprocessing.html#cb1-2" tabindex="-1"></a><span class="co">#SBATCH -p SVC # partition (queue)</span></span>
<span id="cb1-3"><a href="data-preprocessing.html#cb1-3" tabindex="-1"></a><span class="co">#SBATCH --job-name=p53</span></span>
<span id="cb1-4"><a href="data-preprocessing.html#cb1-4" tabindex="-1"></a><span class="co">#SBATCH -n 8</span></span>
<span id="cb1-5"><a href="data-preprocessing.html#cb1-5" tabindex="-1"></a><span class="co">#SBATCH --array=1-10</span></span>
<span id="cb1-6"><a href="data-preprocessing.html#cb1-6" tabindex="-1"></a><span class="co">#SBATCH -t 7-00:00 # time (D-HH:MM)</span></span>
<span id="cb1-7"><a href="data-preprocessing.html#cb1-7" tabindex="-1"></a><span class="co">#SBATCH -o _log/salmon.%N.%A_%a.out # STDOUT</span></span>
<span id="cb1-8"><a href="data-preprocessing.html#cb1-8" tabindex="-1"></a><span class="co">#SBATCH -e _log/salmon.%N.%A_%a.err # STDERR</span></span>
<span id="cb1-9"><a href="data-preprocessing.html#cb1-9" tabindex="-1"></a><span class="co">#SBATCH --mail-type=</span><span class="re">END</span><span class="co">,FAIL # notifications for job done &amp; fail</span></span>
<span id="cb1-10"><a href="data-preprocessing.html#cb1-10" tabindex="-1"></a><span class="co">#SBATCH --mail-user=XX</span></span>
<span id="cb1-11"><a href="data-preprocessing.html#cb1-11" tabindex="-1"></a></span>
<span id="cb1-12"><a href="data-preprocessing.html#cb1-12" tabindex="-1"></a>id<span class="ot">=</span><span class="st">`</span><span class="at">sed -n ${SLURM_ARRAY_TASK_ID}p sample.txt</span><span class="st">`</span></span>
<span id="cb1-13"><a href="data-preprocessing.html#cb1-13" tabindex="-1"></a>echo <span class="st">&quot;${id}&quot;</span></span>
<span id="cb1-14"><a href="data-preprocessing.html#cb1-14" tabindex="-1"></a></span>
<span id="cb1-15"><a href="data-preprocessing.html#cb1-15" tabindex="-1"></a>fq_path<span class="ot">=</span><span class="er">$</span>{path_to_fastq}</span>
<span id="cb1-16"><a href="data-preprocessing.html#cb1-16" tabindex="-1"></a></span>
<span id="cb1-17"><a href="data-preprocessing.html#cb1-17" tabindex="-1"></a>fq1<span class="ot">=</span><span class="er">$</span>{fq_path}<span class="sc">/</span><span class="er">$</span>{id}_R1.fq.gz</span>
<span id="cb1-18"><a href="data-preprocessing.html#cb1-18" tabindex="-1"></a>fq2<span class="ot">=</span><span class="er">$</span>{fq_path}<span class="sc">/</span><span class="er">$</span>{id}_R2.fq.gz</span>
<span id="cb1-19"><a href="data-preprocessing.html#cb1-19" tabindex="-1"></a></span>
<span id="cb1-20"><a href="data-preprocessing.html#cb1-20" tabindex="-1"></a>gtf_file<span class="ot">=</span><span class="er">$</span>{path_to_reference}<span class="sc">/</span>gencode.v40.annotation.gtf</span>
<span id="cb1-21"><a href="data-preprocessing.html#cb1-21" tabindex="-1"></a>salmon_index<span class="ot">=</span><span class="er">$</span>{path_to_reference}<span class="sc">/</span>gencode.v40_salmon<span class="fl">-1.8</span></span>
<span id="cb1-22"><a href="data-preprocessing.html#cb1-22" tabindex="-1"></a>out_path<span class="ot">=</span><span class="er">$</span>{out_path}</span>
<span id="cb1-23"><a href="data-preprocessing.html#cb1-23" tabindex="-1"></a></span>
<span id="cb1-24"><a href="data-preprocessing.html#cb1-24" tabindex="-1"></a><span class="sc">$</span>SALMON_1_8 quant <span class="sc">-</span>p <span class="dv">8</span> <span class="sc">-</span>l IU <span class="sc">-</span>i <span class="sc">$</span>{salmon_index} <span class="sc">-</span>o <span class="sc">$</span>{out_path}<span class="sc">/</span><span class="er">$</span>{id} <span class="sc">-</span><span class="dv">1</span> <span class="sc">$</span>{fq1} <span class="sc">-</span><span class="dv">2</span> <span class="sc">$</span>{fq2} <span class="sc">-</span>g <span class="sc">$</span>{gtf_file} <span class="sc">--</span>gcBias <span class="sc">--</span>validateMappings</span></code></pre></div>
<p>The raw expression of RNA-seq data generated using salmon could be downloaded via <a href="https://github.com/NRCTM-bioinfo/p53_compounds" class="uri">https://github.com/NRCTM-bioinfo/p53_compounds</a>. The <code>Log2(FPKM + 1)</code> was used for evaluation.</p>
</div>
<div id="geo-data" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> GEO data<a href="data-preprocessing.html#geo-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>GEOquery</code> was used for downloading of GEO-based expression data.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="data-preprocessing.html#cb2-1" tabindex="-1"></a><span class="co"># Download GEO data using GEOquery</span></span>
<span id="cb2-2"><a href="data-preprocessing.html#cb2-2" tabindex="-1"></a><span class="co"># Taking GSE15658 as an example</span></span>
<span id="cb2-3"><a href="data-preprocessing.html#cb2-3" tabindex="-1"></a>seq.name.list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;GSE15658&quot;</span>)</span>
<span id="cb2-4"><a href="data-preprocessing.html#cb2-4" tabindex="-1"></a></span>
<span id="cb2-5"><a href="data-preprocessing.html#cb2-5" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(seq.name.list)) {</span>
<span id="cb2-6"><a href="data-preprocessing.html#cb2-6" tabindex="-1"></a>seq.name <span class="ot">=</span> seq.name.list[i]</span>
<span id="cb2-7"><a href="data-preprocessing.html#cb2-7" tabindex="-1"></a><span class="fu">message</span>(i , <span class="st">&quot;   &quot;</span>, seq.name)</span>
<span id="cb2-8"><a href="data-preprocessing.html#cb2-8" tabindex="-1"></a></span>
<span id="cb2-9"><a href="data-preprocessing.html#cb2-9" tabindex="-1"></a>gset <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(seq.name, <span class="at">GSEMatrix =</span> <span class="cn">TRUE</span>, <span class="at">AnnotGPL =</span> <span class="cn">TRUE</span>, <span class="at">getGPL =</span> <span class="cn">TRUE</span> )</span>
<span id="cb2-10"><a href="data-preprocessing.html#cb2-10" tabindex="-1"></a><span class="fu">save</span>(gset, <span class="at">file =</span> <span class="fu">paste0</span>( seq.name, <span class="st">&quot;.geo.rds&quot;</span>))</span>
<span id="cb2-11"><a href="data-preprocessing.html#cb2-11" tabindex="-1"></a>}</span>
<span id="cb2-12"><a href="data-preprocessing.html#cb2-12" tabindex="-1"></a></span>
<span id="cb2-13"><a href="data-preprocessing.html#cb2-13" tabindex="-1"></a></span>
<span id="cb2-14"><a href="data-preprocessing.html#cb2-14" tabindex="-1"></a><span class="co"># load GEO and and perform average intensity </span></span>
<span id="cb2-15"><a href="data-preprocessing.html#cb2-15" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">paste0</span>(<span class="st">&quot;geo/GSE15658.geo.rds&quot;</span>))</span>
<span id="cb2-16"><a href="data-preprocessing.html#cb2-16" tabindex="-1"></a></span>
<span id="cb2-17"><a href="data-preprocessing.html#cb2-17" tabindex="-1"></a>exprset <span class="ot">&lt;-</span> <span class="fu">exprs</span>(gset[[<span class="dv">1</span>]])</span>
<span id="cb2-18"><a href="data-preprocessing.html#cb2-18" tabindex="-1"></a>pdata <span class="ot">&lt;-</span> <span class="fu">pData</span>(gset[[<span class="dv">1</span>]])</span>
<span id="cb2-19"><a href="data-preprocessing.html#cb2-19" tabindex="-1"></a>fdata <span class="ot">&lt;-</span> <span class="fu">fData</span>(gset[[<span class="dv">1</span>]])</span>
<span id="cb2-20"><a href="data-preprocessing.html#cb2-20" tabindex="-1"></a></span>
<span id="cb2-21"><a href="data-preprocessing.html#cb2-21" tabindex="-1"></a>exprset <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(exprset)</span>
<span id="cb2-22"><a href="data-preprocessing.html#cb2-22" tabindex="-1"></a>fdata <span class="ot">&lt;-</span> fdata[<span class="fu">match</span>(<span class="fu">rownames</span>(exprset), fdata[, <span class="dv">1</span>]), ]</span>
<span id="cb2-23"><a href="data-preprocessing.html#cb2-23" tabindex="-1"></a>exprset <span class="ot">&lt;-</span> exprset <span class="sc">-</span> <span class="fu">min</span>(exprset)</span>
<span id="cb2-24"><a href="data-preprocessing.html#cb2-24" tabindex="-1"></a></span>
<span id="cb2-25"><a href="data-preprocessing.html#cb2-25" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">row.names</span>(exprset) <span class="sc">!=</span> fdata[, <span class="st">&quot;ID&quot;</span>])</span>
<span id="cb2-26"><a href="data-preprocessing.html#cb2-26" tabindex="-1"></a>exp_mat <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(exprset, <span class="fu">list</span>(<span class="at">Symbol =</span> fdata[, <span class="st">&quot;Gene symbol&quot;</span>]), mean)</span>
<span id="cb2-27"><a href="data-preprocessing.html#cb2-27" tabindex="-1"></a>exp_mat <span class="ot">&lt;-</span> exp_mat[exp_mat<span class="sc">$</span>Symbol <span class="sc">!=</span> <span class="st">&quot;&quot;</span>, ]</span>
<span id="cb2-28"><a href="data-preprocessing.html#cb2-28" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat) <span class="ot">&lt;-</span> exp_mat<span class="sc">$</span>Symbol</span>
<span id="cb2-29"><a href="data-preprocessing.html#cb2-29" tabindex="-1"></a>exp_mat<span class="sc">$</span>Symbol <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-30"><a href="data-preprocessing.html#cb2-30" tabindex="-1"></a></span>
<span id="cb2-31"><a href="data-preprocessing.html#cb2-31" tabindex="-1"></a>exp_mat <span class="ot">&lt;-</span> <span class="fu">normalizeBetweenArrays</span>(exp_mat)</span>
<span id="cb2-32"><a href="data-preprocessing.html#cb2-32" tabindex="-1"></a></span>
<span id="cb2-33"><a href="data-preprocessing.html#cb2-33" tabindex="-1"></a></span>
<span id="cb2-34"><a href="data-preprocessing.html#cb2-34" tabindex="-1"></a><span class="co"># For the dataset GSE137574, we used anovar to perform annotation.</span></span>
<span id="cb2-35"><a href="data-preprocessing.html#cb2-35" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">paste0</span>(<span class="st">&quot;geo/GSE137574.geo.rds&quot;</span>))</span>
<span id="cb2-36"><a href="data-preprocessing.html#cb2-36" tabindex="-1"></a></span>
<span id="cb2-37"><a href="data-preprocessing.html#cb2-37" tabindex="-1"></a>exprset <span class="ot">&lt;-</span> <span class="fu">exprs</span>(gset[[<span class="dv">1</span>]])</span>
<span id="cb2-38"><a href="data-preprocessing.html#cb2-38" tabindex="-1"></a>pdata <span class="ot">&lt;-</span> <span class="fu">pData</span>(gset[[<span class="dv">1</span>]])</span>
<span id="cb2-39"><a href="data-preprocessing.html#cb2-39" tabindex="-1"></a>fdata <span class="ot">&lt;-</span> <span class="fu">fData</span>(gset[[<span class="dv">1</span>]])</span>
<span id="cb2-40"><a href="data-preprocessing.html#cb2-40" tabindex="-1"></a></span>
<span id="cb2-41"><a href="data-preprocessing.html#cb2-41" tabindex="-1"></a>exprset <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(exprset)</span>
<span id="cb2-42"><a href="data-preprocessing.html#cb2-42" tabindex="-1"></a>fdata <span class="ot">&lt;-</span> fdata[<span class="fu">match</span>(<span class="fu">rownames</span>(exprset), fdata[, <span class="dv">1</span>]), ]</span>
<span id="cb2-43"><a href="data-preprocessing.html#cb2-43" tabindex="-1"></a>exprset <span class="ot">&lt;-</span> exprset <span class="sc">-</span> <span class="fu">min</span>(exprset)</span>
<span id="cb2-44"><a href="data-preprocessing.html#cb2-44" tabindex="-1"></a></span>
<span id="cb2-45"><a href="data-preprocessing.html#cb2-45" tabindex="-1"></a><span class="fu">dim</span>(fdata[<span class="fu">which</span>(fdata<span class="sc">$</span>unigene <span class="sc">!=</span> <span class="st">&quot;---&quot;</span>), ])</span>
<span id="cb2-46"><a href="data-preprocessing.html#cb2-46" tabindex="-1"></a>out.data <span class="ot">&lt;-</span> fdata[<span class="fu">which</span>(fdata<span class="sc">$</span>unigene <span class="sc">!=</span> <span class="st">&quot;---&quot;</span>), ]</span>
<span id="cb2-47"><a href="data-preprocessing.html#cb2-47" tabindex="-1"></a>out.data <span class="ot">&lt;-</span> out.data[, <span class="fu">c</span>(<span class="st">&quot;seqname&quot;</span>,<span class="st">&quot;start&quot;</span>,<span class="st">&quot;stop&quot;</span>,<span class="st">&quot;strand&quot;</span>,<span class="st">&quot;probeset_id&quot;</span>,<span class="st">&quot;ID&quot;</span>)]</span>
<span id="cb2-48"><a href="data-preprocessing.html#cb2-48" tabindex="-1"></a>out.data<span class="sc">$</span>strand <span class="ot">=</span> <span class="st">&quot;0&quot;</span></span>
<span id="cb2-49"><a href="data-preprocessing.html#cb2-49" tabindex="-1"></a>out.data<span class="sc">$</span>probeset_id <span class="ot">=</span> <span class="st">&quot;-&quot;</span></span>
<span id="cb2-50"><a href="data-preprocessing.html#cb2-50" tabindex="-1"></a><span class="fu">write.table</span>(out.data, <span class="st">&quot;geo/GSE137574.avinput&quot;</span>, <span class="at">row.names =</span> F, <span class="at">col.names =</span> F, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> F)</span>
<span id="cb2-51"><a href="data-preprocessing.html#cb2-51" tabindex="-1"></a></span>
<span id="cb2-52"><a href="data-preprocessing.html#cb2-52" tabindex="-1"></a><span class="st">&quot;table_annovar.pl GSE137574.avinput /public/home/daiyt/bin/annovar/annovar/humandb -buildver hg19 -out GSE137574.anno  -remove -protocol refGene -operation g -nastring . &quot;</span></span>
<span id="cb2-53"><a href="data-preprocessing.html#cb2-53" tabindex="-1"></a></span>
<span id="cb2-54"><a href="data-preprocessing.html#cb2-54" tabindex="-1"></a>out.data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;geo/GSE137574.anno.hg19_multianno.txt&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb2-55"><a href="data-preprocessing.html#cb2-55" tabindex="-1"></a>out.data<span class="sc">$</span>Gene <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(out.data<span class="sc">$</span>Gene.refGene, <span class="fu">regex</span>(<span class="st">&quot;;.+&quot;</span>), <span class="st">&quot;&quot;</span>)</span>
<span id="cb2-56"><a href="data-preprocessing.html#cb2-56" tabindex="-1"></a></span>
<span id="cb2-57"><a href="data-preprocessing.html#cb2-57" tabindex="-1"></a>fdata <span class="ot">&lt;-</span> fdata[<span class="fu">which</span>(fdata<span class="sc">$</span>unigene <span class="sc">!=</span> <span class="st">&quot;---&quot;</span>), ]</span>
<span id="cb2-58"><a href="data-preprocessing.html#cb2-58" tabindex="-1"></a><span class="fu">sum</span>(fdata<span class="sc">$</span>start <span class="sc">!=</span> out.data<span class="sc">$</span>Start)</span>
<span id="cb2-59"><a href="data-preprocessing.html#cb2-59" tabindex="-1"></a>exprset <span class="ot">&lt;-</span> exprset[fdata<span class="sc">$</span>ID, ]</span>
<span id="cb2-60"><a href="data-preprocessing.html#cb2-60" tabindex="-1"></a>fdata<span class="sc">$</span>GeneSymbol <span class="ot">&lt;-</span> out.data<span class="sc">$</span>Gene</span>
<span id="cb2-61"><a href="data-preprocessing.html#cb2-61" tabindex="-1"></a></span>
<span id="cb2-62"><a href="data-preprocessing.html#cb2-62" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">row.names</span>(exprset) <span class="sc">!=</span> fdata[, <span class="st">&quot;ID&quot;</span>])</span>
<span id="cb2-63"><a href="data-preprocessing.html#cb2-63" tabindex="-1"></a>exp_mat <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(exprset, <span class="fu">list</span>(<span class="at">Symbol =</span> fdata[, <span class="st">&quot;GeneSymbol&quot;</span>]), mean)</span>
<span id="cb2-64"><a href="data-preprocessing.html#cb2-64" tabindex="-1"></a>exp_mat <span class="ot">&lt;-</span> exp_mat[exp_mat<span class="sc">$</span>Symbol <span class="sc">!=</span> <span class="st">&quot;&quot;</span>, ]</span>
<span id="cb2-65"><a href="data-preprocessing.html#cb2-65" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat) <span class="ot">&lt;-</span> exp_mat<span class="sc">$</span>Symbol</span>
<span id="cb2-66"><a href="data-preprocessing.html#cb2-66" tabindex="-1"></a>exp_mat<span class="sc">$</span>Symbol <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb2-67"><a href="data-preprocessing.html#cb2-67" tabindex="-1"></a></span>
<span id="cb2-68"><a href="data-preprocessing.html#cb2-68" tabindex="-1"></a>exp_mat <span class="ot">&lt;-</span> <span class="fu">log2</span>(exp_mat)</span></code></pre></div>
<p>The raw data of array-based GEO data could be downloaded via <a href="https://github.com/NRCTM-bioinfo/p53_compounds" class="uri">https://github.com/NRCTM-bioinfo/p53_compounds</a>.</p>
</div>
<div id="source-data" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Source data<a href="data-preprocessing.html#source-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>One dataset (COTI-2 compound using RNA-seq) was downloaded directly from the supplementary data provided by the original article.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="data-preprocessing.html#cb3-1" tabindex="-1"></a>plot.data <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="st">&quot;COTI-2.xlsx&quot;</span>)</span>
<span id="cb3-2"><a href="data-preprocessing.html#cb3-2" tabindex="-1"></a>plot.data <span class="ot">&lt;-</span> plot.data[<span class="fu">match</span>(<span class="fu">unique</span>(plot.data<span class="sc">$</span>Symbol), plot.data<span class="sc">$</span>Symbol), ]</span>
<span id="cb3-3"><a href="data-preprocessing.html#cb3-3" tabindex="-1"></a>plot.data <span class="ot">&lt;-</span> plot.data[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(plot.data<span class="sc">$</span>Symbol)), ]</span>
<span id="cb3-4"><a href="data-preprocessing.html#cb3-4" tabindex="-1"></a>plot.mat <span class="ot">&lt;-</span> plot.data[, <span class="dv">7</span><span class="sc">:</span><span class="dv">30</span>]</span>
<span id="cb3-5"><a href="data-preprocessing.html#cb3-5" tabindex="-1"></a><span class="fu">rownames</span>(plot.mat) <span class="ot">&lt;-</span> plot.data<span class="sc">$</span>Symbol</span>
<span id="cb3-6"><a href="data-preprocessing.html#cb3-6" tabindex="-1"></a></span>
<span id="cb3-7"><a href="data-preprocessing.html#cb3-7" tabindex="-1"></a>countToFpkm <span class="ot">&lt;-</span> <span class="cf">function</span>(count, geneLength){</span>
<span id="cb3-8"><a href="data-preprocessing.html#cb3-8" tabindex="-1"></a>  fpkm <span class="ot">&lt;-</span> <span class="fu">apply</span>(count, <span class="dv">2</span>, <span class="cf">function</span>(e){ e<span class="sc">*</span><span class="dv">1000</span><span class="sc">*</span><span class="dv">1000000</span><span class="sc">/</span><span class="fu">sum</span>(e)<span class="sc">/</span>geneLength })</span>
<span id="cb3-9"><a href="data-preprocessing.html#cb3-9" tabindex="-1"></a>  <span class="fu">return</span>(fpkm)</span>
<span id="cb3-10"><a href="data-preprocessing.html#cb3-10" tabindex="-1"></a>}</span>
<span id="cb3-11"><a href="data-preprocessing.html#cb3-11" tabindex="-1"></a></span>
<span id="cb3-12"><a href="data-preprocessing.html#cb3-12" tabindex="-1"></a>geneLength <span class="ot">&lt;-</span> plot.data<span class="sc">$</span>TotalExonLength</span>
<span id="cb3-13"><a href="data-preprocessing.html#cb3-13" tabindex="-1"></a><span class="fu">names</span>(geneLength) <span class="ot">&lt;-</span> plot.data<span class="sc">$</span>Symbol</span>
<span id="cb3-14"><a href="data-preprocessing.html#cb3-14" tabindex="-1"></a>exp_mat <span class="ot">&lt;-</span> <span class="fu">log2</span>(<span class="fu">countToFpkm</span>(<span class="fu">as.matrix</span>(plot.mat), geneLength) <span class="sc">+</span> <span class="dv">1</span>)</span></code></pre></div>
<p>The raw data of this dataset could be downloaded via <a href="https://github.com/NRCTM-bioinfo/p53_compounds" class="uri">https://github.com/NRCTM-bioinfo/p53_compounds</a>.</p>
<p>Gene Symbol Converter</p>
<p>We noticed some gene symbol for several datasets was old version, here we provided the source code for gene symbol transformation.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="data-preprocessing.html#cb4-1" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;WDR63&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;DNAI3&quot;</span></span>
<span id="cb4-2"><a href="data-preprocessing.html#cb4-2" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;FAM198B&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;GASK1B&quot;</span></span>
<span id="cb4-3"><a href="data-preprocessing.html#cb4-3" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;FAM212B&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;INKA2&quot;</span></span>
<span id="cb4-4"><a href="data-preprocessing.html#cb4-4" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;KIAA1456&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;TRMT9B&quot;</span></span>
<span id="cb4-5"><a href="data-preprocessing.html#cb4-5" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;LINC01314&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;CTXND1&quot;</span></span>
<span id="cb4-6"><a href="data-preprocessing.html#cb4-6" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;MARC2&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;MTARC2&quot;</span></span>
<span id="cb4-7"><a href="data-preprocessing.html#cb4-7" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;C11orf63&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;JHY&quot;</span></span>
<span id="cb4-8"><a href="data-preprocessing.html#cb4-8" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;LINC00959&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;C10orf143&quot;</span></span>
<span id="cb4-9"><a href="data-preprocessing.html#cb4-9" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;FAM155B&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;NALF2&quot;</span></span>
<span id="cb4-10"><a href="data-preprocessing.html#cb4-10" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;FAM46C&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;TENT5C&quot;</span></span>
<span id="cb4-11"><a href="data-preprocessing.html#cb4-11" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;FAM19A1&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;TAFA1&quot;</span></span>
<span id="cb4-12"><a href="data-preprocessing.html#cb4-12" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;C9orf171&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;CFAP77&quot;</span></span>
<span id="cb4-13"><a href="data-preprocessing.html#cb4-13" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;C1orf168&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;FYB2&quot;</span></span>
<span id="cb4-14"><a href="data-preprocessing.html#cb4-14" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;ARSE&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;ARSL&quot;</span></span>
<span id="cb4-15"><a href="data-preprocessing.html#cb4-15" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;SPERT&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;CBY2&quot;</span></span>
<span id="cb4-16"><a href="data-preprocessing.html#cb4-16" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST1H2AJ&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H2AC14&quot;</span></span>
<span id="cb4-17"><a href="data-preprocessing.html#cb4-17" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST1H1D&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H1-3&quot;</span></span>
<span id="cb4-18"><a href="data-preprocessing.html#cb4-18" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST1H2BO&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H2BC17&quot;</span></span>
<span id="cb4-19"><a href="data-preprocessing.html#cb4-19" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST1H2AE&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H2AC8&quot;</span></span>
<span id="cb4-20"><a href="data-preprocessing.html#cb4-20" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST3H2BB&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H2BC26&quot;</span></span>
<span id="cb4-21"><a href="data-preprocessing.html#cb4-21" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST1H3J&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H3C12&quot;</span></span>
<span id="cb4-22"><a href="data-preprocessing.html#cb4-22" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST1H2AI&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H2AC13&quot;</span></span>
<span id="cb4-23"><a href="data-preprocessing.html#cb4-23" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST1H2AG&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H2AC11&quot;</span></span>
<span id="cb4-24"><a href="data-preprocessing.html#cb4-24" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST1H2BK&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H2BC12&quot;</span></span>
<span id="cb4-25"><a href="data-preprocessing.html#cb4-25" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST1H4H&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H4C8&quot;</span></span>
<span id="cb4-26"><a href="data-preprocessing.html#cb4-26" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST1H2BJ&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H2BC11&quot;</span></span>
<span id="cb4-27"><a href="data-preprocessing.html#cb4-27" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST1H2AM&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H2AC17&quot;</span></span>
<span id="cb4-28"><a href="data-preprocessing.html#cb4-28" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST1H3H&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H3C10&quot;</span></span>
<span id="cb4-29"><a href="data-preprocessing.html#cb4-29" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;CCDC163P&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;CCDC163&quot;</span></span>
<span id="cb4-30"><a href="data-preprocessing.html#cb4-30" tabindex="-1"></a><span class="fu">rownames</span>(exp_mat)[<span class="fu">which</span>(<span class="fu">rownames</span>(exp_mat) <span class="sc">==</span> <span class="st">&quot;HIST1H2BL&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;H2BC13&quot;</span></span></code></pre></div>
</div>
<div id="tcga-data" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> TCGA data<a href="data-preprocessing.html#tcga-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The gene expression data and clinical data of the TCGA Pan-Cancer cohorts included in this study can be accessed through the following link: <a href="https://xenabrowser.net/datapages/" class="uri">https://xenabrowser.net/datapages/</a></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="target-evaluation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/01-DataPreprocessing.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["p53-compounds.pdf", "p53-compounds.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
