<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 scRNA-seq | APL stemness</title>
  <meta name="description" content="This is the tutorial of APL-stemness project." />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 scRNA-seq | APL stemness" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://nrctm-bioinfo.github.io/APL_stemness/index.html/figure/cover.jpg" />
  <meta property="og:description" content="This is the tutorial of APL-stemness project." />
  <meta name="github-repo" content="NRCTM-bioinfo/APL-stemness" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 scRNA-seq | APL stemness" />
  
  <meta name="twitter:description" content="This is the tutorial of APL-stemness project." />
  <meta name="twitter:image" content="https://nrctm-bioinfo.github.io/APL_stemness/index.html/figure/cover.jpg" />

<meta name="author" content="Yu-Ting Dai, Wen Jin" />


<meta name="date" content="2023-11-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="bulk.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://github.com/NRCTM-bioinfo/APL-stemness"><center><b>APL-stemness</b></center></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="scrna-seq.html"><a href="scrna-seq.html"><i class="fa fa-check"></i><b>2</b> scRNA-seq</a>
<ul>
<li class="chapter" data-level="2.1" data-path="scrna-seq.html"><a href="scrna-seq.html#load-packages"><i class="fa fa-check"></i><b>2.1</b> Load packages</a></li>
<li class="chapter" data-level="2.2" data-path="scrna-seq.html"><a href="scrna-seq.html#pre-processing-of-raw-data"><i class="fa fa-check"></i><b>2.2</b> Pre-processing of raw data</a></li>
<li class="chapter" data-level="2.3" data-path="scrna-seq.html"><a href="scrna-seq.html#normal-bmmcs"><i class="fa fa-check"></i><b>2.3</b> Normal BMMCs</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="scrna-seq.html"><a href="scrna-seq.html#pre-processing-of-bmmcs"><i class="fa fa-check"></i><b>2.3.1</b> Pre-processing of BMMCs</a></li>
<li class="chapter" data-level="2.3.2" data-path="scrna-seq.html"><a href="scrna-seq.html#annotation-of-normal-bmmcs"><i class="fa fa-check"></i><b>2.3.2</b> Annotation of normal BMMCs</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="scrna-seq.html"><a href="scrna-seq.html#apls"><i class="fa fa-check"></i><b>2.4</b> APLs</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="scrna-seq.html"><a href="scrna-seq.html#pre-processing-of-apls"><i class="fa fa-check"></i><b>2.4.1</b> Pre-processing of APLs</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="scrna-seq.html"><a href="scrna-seq.html#integration"><i class="fa fa-check"></i><b>2.5</b> Integration</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="scrna-seq.html"><a href="scrna-seq.html#integration-of-apls-and-normal-bmmcs"><i class="fa fa-check"></i><b>2.5.1</b> Integration of APLs and normal BMMCs</a></li>
<li class="chapter" data-level="2.5.2" data-path="scrna-seq.html"><a href="scrna-seq.html#normalization"><i class="fa fa-check"></i><b>2.5.2</b> Normalization</a></li>
<li class="chapter" data-level="2.5.3" data-path="scrna-seq.html"><a href="scrna-seq.html#batch-effect-adjustment"><i class="fa fa-check"></i><b>2.5.3</b> Batch effect adjustment</a></li>
<li class="chapter" data-level="2.5.4" data-path="scrna-seq.html"><a href="scrna-seq.html#cell-annotation"><i class="fa fa-check"></i><b>2.5.4</b> Cell annotation</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="scrna-seq.html"><a href="scrna-seq.html#apl-blasts"><i class="fa fa-check"></i><b>2.6</b> APL blasts</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="scrna-seq.html"><a href="scrna-seq.html#subset-apl-blasts"><i class="fa fa-check"></i><b>2.6.1</b> Subset APL blasts</a></li>
<li class="chapter" data-level="2.6.2" data-path="scrna-seq.html"><a href="scrna-seq.html#normalization-1"><i class="fa fa-check"></i><b>2.6.2</b> Normalization</a></li>
<li class="chapter" data-level="2.6.3" data-path="scrna-seq.html"><a href="scrna-seq.html#batch-effect-adjustment-of-apl-blasts"><i class="fa fa-check"></i><b>2.6.3</b> Batch effect adjustment of APL blasts</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="scrna-seq.html"><a href="scrna-seq.html#viper"><i class="fa fa-check"></i><b>2.7</b> VIPER</a></li>
<li class="chapter" data-level="2.8" data-path="scrna-seq.html"><a href="scrna-seq.html#dorothea"><i class="fa fa-check"></i><b>2.8</b> dorothea</a></li>
<li class="chapter" data-level="2.9" data-path="scrna-seq.html"><a href="scrna-seq.html#cell-signature"><i class="fa fa-check"></i><b>2.9</b> Cell signature</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="bulk.html"><a href="bulk.html"><i class="fa fa-check"></i><b>3</b> Bulk</a>
<ul>
<li class="chapter" data-level="3.1" data-path="bulk.html"><a href="bulk.html#pre-processing"><i class="fa fa-check"></i><b>3.1</b> Pre-processing</a></li>
<li class="chapter" data-level="3.2" data-path="bulk.html"><a href="bulk.html#degs"><i class="fa fa-check"></i><b>3.2</b> DEGs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="deconvolution.html"><a href="deconvolution.html"><i class="fa fa-check"></i><b>4</b> Deconvolution</a>
<ul>
<li class="chapter" data-level="4.1" data-path="deconvolution.html"><a href="deconvolution.html#svr"><i class="fa fa-check"></i><b>4.1</b> SVR</a></li>
<li class="chapter" data-level="4.2" data-path="deconvolution.html"><a href="deconvolution.html#regression"><i class="fa fa-check"></i><b>4.2</b> Regression</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="about-this-project.html"><a href="about-this-project.html"><i class="fa fa-check"></i><b>5</b> About this project</a>
<ul>
<li class="chapter" data-level="" data-path="about-this-project.html"><a href="about-this-project.html#current-version"><i class="fa fa-check"></i>Current version</a></li>
<li class="chapter" data-level="" data-path="about-this-project.html"><a href="about-this-project.html#session-information"><i class="fa fa-check"></i>Session Information</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">APL stemness</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="scrna-seq" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> scRNA-seq<a href="scrna-seq.html#scrna-seq" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="load-packages" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Load packages<a href="scrna-seq.html#load-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="scrna-seq.html#cb1-1" tabindex="-1"></a><span class="fu">suppressMessages</span>({</span>
<span id="cb1-2"><a href="scrna-seq.html#cb1-2" tabindex="-1"></a>  <span class="fu">library</span>(dplyr)</span>
<span id="cb1-3"><a href="scrna-seq.html#cb1-3" tabindex="-1"></a>  <span class="fu">library</span>(e1071)</span>
<span id="cb1-4"><a href="scrna-seq.html#cb1-4" tabindex="-1"></a>  <span class="fu">library</span>(pheatmap)</span>
<span id="cb1-5"><a href="scrna-seq.html#cb1-5" tabindex="-1"></a>  <span class="fu">library</span>(ggplot2)</span>
<span id="cb1-6"><a href="scrna-seq.html#cb1-6" tabindex="-1"></a>  <span class="fu">library</span>(DESeq2)</span>
<span id="cb1-7"><a href="scrna-seq.html#cb1-7" tabindex="-1"></a>  <span class="fu">library</span>(openxlsx)</span>
<span id="cb1-8"><a href="scrna-seq.html#cb1-8" tabindex="-1"></a>  <span class="fu">library</span>(dendextend)</span>
<span id="cb1-9"><a href="scrna-seq.html#cb1-9" tabindex="-1"></a>  <span class="fu">library</span>(matrixStats)</span>
<span id="cb1-10"><a href="scrna-seq.html#cb1-10" tabindex="-1"></a>  <span class="fu">library</span>(ggpubr)</span>
<span id="cb1-11"><a href="scrna-seq.html#cb1-11" tabindex="-1"></a>  <span class="fu">library</span>(limma)</span>
<span id="cb1-12"><a href="scrna-seq.html#cb1-12" tabindex="-1"></a>  <span class="fu">library</span>(randomForest)</span>
<span id="cb1-13"><a href="scrna-seq.html#cb1-13" tabindex="-1"></a>  <span class="fu">library</span>(clusterProfiler)</span>
<span id="cb1-14"><a href="scrna-seq.html#cb1-14" tabindex="-1"></a>  <span class="fu">library</span>(genefilter)</span>
<span id="cb1-15"><a href="scrna-seq.html#cb1-15" tabindex="-1"></a>  <span class="fu">library</span>(GSVA)</span>
<span id="cb1-16"><a href="scrna-seq.html#cb1-16" tabindex="-1"></a>  <span class="fu">library</span>(Biobase)</span>
<span id="cb1-17"><a href="scrna-seq.html#cb1-17" tabindex="-1"></a>  <span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb1-18"><a href="scrna-seq.html#cb1-18" tabindex="-1"></a>  <span class="fu">library</span>(ggplot2)</span>
<span id="cb1-19"><a href="scrna-seq.html#cb1-19" tabindex="-1"></a>  <span class="fu">library</span>(ggthemes)</span>
<span id="cb1-20"><a href="scrna-seq.html#cb1-20" tabindex="-1"></a>  <span class="fu">library</span>(enrichplot)</span>
<span id="cb1-21"><a href="scrna-seq.html#cb1-21" tabindex="-1"></a>  <span class="fu">library</span>(GSEABase)</span>
<span id="cb1-22"><a href="scrna-seq.html#cb1-22" tabindex="-1"></a>  <span class="fu">library</span>(Seurat)</span>
<span id="cb1-23"><a href="scrna-seq.html#cb1-23" tabindex="-1"></a>  <span class="fu">library</span>(monocle)</span>
<span id="cb1-24"><a href="scrna-seq.html#cb1-24" tabindex="-1"></a>})</span></code></pre></div>
</div>
<div id="pre-processing-of-raw-data" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Pre-processing of raw data<a href="scrna-seq.html#pre-processing-of-raw-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="scrna-seq.html#cb2-1" tabindex="-1"></a><span class="co"># Pre-processing APL and normal BMMCs using cellranger</span></span>
<span id="cb2-2"><a href="scrna-seq.html#cb2-2" tabindex="-1"></a>cellranger count <span class="sc">--</span>id<span class="ot">=</span><span class="er">$</span>{Sample_ID} \</span>
<span id="cb2-3"><a href="scrna-seq.html#cb2-3" tabindex="-1"></a>  <span class="sc">--</span>fastqs<span class="ot">=</span><span class="er">$</span>{FASTQ} \</span>
<span id="cb2-4"><a href="scrna-seq.html#cb2-4" tabindex="-1"></a>  <span class="sc">--</span>sample<span class="ot">=</span><span class="er">$</span>{Sample_ID}\</span>
<span id="cb2-5"><a href="scrna-seq.html#cb2-5" tabindex="-1"></a>  <span class="sc">--</span>transcriptome<span class="ot">=</span><span class="er">$</span>{index_path}</span></code></pre></div>
</div>
<div id="normal-bmmcs" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Normal BMMCs<a href="scrna-seq.html#normal-bmmcs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="pre-processing-of-bmmcs" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Pre-processing of BMMCs<a href="scrna-seq.html#pre-processing-of-bmmcs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="scrna-seq.html#cb3-1" tabindex="-1"></a>bmmc <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> bmmc.data, <span class="at">project =</span> <span class="st">&quot;BMMC&quot;</span>, <span class="at">min.cells =</span> <span class="dv">0</span>, <span class="at">min.features =</span> <span class="dv">0</span>)</span>
<span id="cb3-2"><a href="scrna-seq.html#cb3-2" tabindex="-1"></a>bmmc[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(bmmc, <span class="at">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</span>
<span id="cb3-3"><a href="scrna-seq.html#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="scrna-seq.html#cb3-4" tabindex="-1"></a><span class="co"># Pre-processing data was performed using Seurat</span></span>
<span id="cb3-5"><a href="scrna-seq.html#cb3-5" tabindex="-1"></a>bmmc <span class="ot">&lt;-</span> <span class="fu">subset</span>(bmmc, <span class="at">subset =</span> nFeature_RNA <span class="sc">&gt;</span> <span class="dv">300</span> <span class="sc">&amp;</span> percent.mt <span class="sc">&lt;</span> <span class="dv">10</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">30000</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&gt;</span> <span class="dv">500</span> <span class="sc">&amp;</span> Doublet <span class="sc">==</span> <span class="cn">FALSE</span> )</span>
<span id="cb3-6"><a href="scrna-seq.html#cb3-6" tabindex="-1"></a></span>
<span id="cb3-7"><a href="scrna-seq.html#cb3-7" tabindex="-1"></a></span>
<span id="cb3-8"><a href="scrna-seq.html#cb3-8" tabindex="-1"></a>bmmc <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(bmmc, <span class="at">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>, <span class="at">scale.factor =</span> <span class="fu">ncol</span>(bmmc) )</span>
<span id="cb3-9"><a href="scrna-seq.html#cb3-9" tabindex="-1"></a>bmmc <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(bmmc, <span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">3000</span>)</span>
<span id="cb3-10"><a href="scrna-seq.html#cb3-10" tabindex="-1"></a></span>
<span id="cb3-11"><a href="scrna-seq.html#cb3-11" tabindex="-1"></a>bmmc <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(bmmc, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(bmmc))</span>
<span id="cb3-12"><a href="scrna-seq.html#cb3-12" tabindex="-1"></a>bmmc <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(bmmc, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> bmmc))</span>
<span id="cb3-13"><a href="scrna-seq.html#cb3-13" tabindex="-1"></a>bmmc <span class="ot">&lt;-</span> <span class="fu">RunHarmony</span>(bmmc, <span class="st">&quot;Sample&quot;</span>)</span>
<span id="cb3-14"><a href="scrna-seq.html#cb3-14" tabindex="-1"></a>bmmc <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(bmmc, <span class="at">reduction =</span> <span class="st">&quot;harmony&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb3-15"><a href="scrna-seq.html#cb3-15" tabindex="-1"></a>bmmc <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(bmmc, <span class="at">resolution =</span> <span class="fl">0.8</span>)</span>
<span id="cb3-16"><a href="scrna-seq.html#cb3-16" tabindex="-1"></a>bmmc <span class="ot">&lt;-</span> <span class="fu">RunTSNE</span>(bmmc, <span class="at">reduction =</span> <span class="st">&quot;harmony&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">perplexity =</span> <span class="dv">50</span>)</span>
<span id="cb3-17"><a href="scrna-seq.html#cb3-17" tabindex="-1"></a>bmmc <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(bmmc, <span class="at">reduction =</span> <span class="st">&quot;harmony&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb3-18"><a href="scrna-seq.html#cb3-18" tabindex="-1"></a></span>
<span id="cb3-19"><a href="scrna-seq.html#cb3-19" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(bmmc, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">pt.size =</span> <span class="dv">1</span>, </span>
<span id="cb3-20"><a href="scrna-seq.html#cb3-20" tabindex="-1"></a>             <span class="at">raster =</span> <span class="cn">FALSE</span>, <span class="at">cols =</span> color.lib) <span class="sc">+</span> <span class="fu">theme_few</span>()</span>
<span id="cb3-21"><a href="scrna-seq.html#cb3-21" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out.path, <span class="st">&quot;/3.ctrl.cluster.pdf&quot;</span>), p, <span class="at">width =</span> <span class="dv">9</span>, <span class="at">height =</span> <span class="dv">7</span>)</span>
<span id="cb3-22"><a href="scrna-seq.html#cb3-22" tabindex="-1"></a></span>
<span id="cb3-23"><a href="scrna-seq.html#cb3-23" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(bmmc, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">pt.size =</span> <span class="dv">1</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">label.size =</span> <span class="dv">10</span>,</span>
<span id="cb3-24"><a href="scrna-seq.html#cb3-24" tabindex="-1"></a>             <span class="at">raster =</span> <span class="cn">FALSE</span>, <span class="at">cols =</span> color.lib) <span class="sc">+</span> <span class="fu">theme_few</span>()</span>
<span id="cb3-25"><a href="scrna-seq.html#cb3-25" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out.path, <span class="st">&quot;/3.ctrl.cluster.label.pdf&quot;</span>), p, <span class="at">width =</span> <span class="dv">9</span>, <span class="at">height =</span> <span class="dv">7</span>)</span></code></pre></div>
</div>
<div id="annotation-of-normal-bmmcs" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Annotation of normal BMMCs<a href="scrna-seq.html#annotation-of-normal-bmmcs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="scrna-seq.html#cb4-1" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;SingleR/hs.BlueprintEncodeData.RDS&quot;</span>)</span>
<span id="cb4-2"><a href="scrna-seq.html#cb4-2" tabindex="-1"></a>pred.BlueprintEncodeData <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> bmmc<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data, <span class="at">ref =</span> ref, <span class="at">labels =</span> ref<span class="sc">$</span>label.main)</span>
<span id="cb4-3"><a href="scrna-seq.html#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="scrna-seq.html#cb4-4" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;SingleR/hs.HumanPrimaryCellAtlasData.RDS&quot;</span>)</span>
<span id="cb4-5"><a href="scrna-seq.html#cb4-5" tabindex="-1"></a>pred.HumanPrimaryCellAtlasData <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> bmmc<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data, <span class="at">ref =</span> ref, <span class="at">labels =</span> ref<span class="sc">$</span>label.main)</span>
<span id="cb4-6"><a href="scrna-seq.html#cb4-6" tabindex="-1"></a></span>
<span id="cb4-7"><a href="scrna-seq.html#cb4-7" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;SingleR/ImmGenData.RDS&quot;</span>)</span>
<span id="cb4-8"><a href="scrna-seq.html#cb4-8" tabindex="-1"></a>pred.ImmGenData <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> bmmc<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data, <span class="at">ref =</span> ref, <span class="at">labels =</span> ref<span class="sc">$</span>label.main)</span>
<span id="cb4-9"><a href="scrna-seq.html#cb4-9" tabindex="-1"></a></span>
<span id="cb4-10"><a href="scrna-seq.html#cb4-10" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;SingleR/MonacoImmuneData.RDS&quot;</span>)</span>
<span id="cb4-11"><a href="scrna-seq.html#cb4-11" tabindex="-1"></a>pred.MonacoImmuneData <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> bmmc<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data, <span class="at">ref =</span> ref, <span class="at">labels =</span> ref<span class="sc">$</span>label.main)</span>
<span id="cb4-12"><a href="scrna-seq.html#cb4-12" tabindex="-1"></a></span>
<span id="cb4-13"><a href="scrna-seq.html#cb4-13" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;SingleR/NovershternHematopoieticData.RDS&quot;</span>)</span>
<span id="cb4-14"><a href="scrna-seq.html#cb4-14" tabindex="-1"></a>pred.NovershternHematopoieticData <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> bmmc<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data, <span class="at">ref =</span> ref, <span class="at">labels =</span> ref<span class="sc">$</span>label.main)</span></code></pre></div>
</div>
</div>
<div id="apls" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> APLs<a href="scrna-seq.html#apls" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="pre-processing-of-apls" class="section level3 hasAnchor" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Pre-processing of APLs<a href="scrna-seq.html#pre-processing-of-apls" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="scrna-seq.html#cb5-1" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> apl.data, <span class="at">project =</span> <span class="st">&quot;BMMC&quot;</span>, <span class="at">min.cells =</span> <span class="dv">0</span>, <span class="at">min.features =</span> <span class="dv">0</span>)</span>
<span id="cb5-2"><a href="scrna-seq.html#cb5-2" tabindex="-1"></a>apl[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(apl, <span class="at">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</span>
<span id="cb5-3"><a href="scrna-seq.html#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="scrna-seq.html#cb5-4" tabindex="-1"></a><span class="co"># Pre-processing data was performed using Seurat</span></span>
<span id="cb5-5"><a href="scrna-seq.html#cb5-5" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">subset</span>(apl, <span class="at">subset =</span> nFeature_RNA <span class="sc">&gt;</span> <span class="dv">300</span> <span class="sc">&amp;</span> percent.mt <span class="sc">&lt;</span> <span class="dv">10</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&lt;</span> <span class="dv">30000</span> <span class="sc">&amp;</span> nCount_RNA <span class="sc">&gt;</span> <span class="dv">500</span> <span class="sc">&amp;</span> Doublet <span class="sc">==</span> <span class="cn">FALSE</span> )</span>
<span id="cb5-6"><a href="scrna-seq.html#cb5-6" tabindex="-1"></a></span>
<span id="cb5-7"><a href="scrna-seq.html#cb5-7" tabindex="-1"></a></span>
<span id="cb5-8"><a href="scrna-seq.html#cb5-8" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(apl, <span class="at">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>, <span class="at">scale.factor =</span> <span class="fu">ncol</span>(apl) )</span>
<span id="cb5-9"><a href="scrna-seq.html#cb5-9" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(apl, <span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">3000</span>)</span>
<span id="cb5-10"><a href="scrna-seq.html#cb5-10" tabindex="-1"></a></span>
<span id="cb5-11"><a href="scrna-seq.html#cb5-11" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(apl, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(apl))</span>
<span id="cb5-12"><a href="scrna-seq.html#cb5-12" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(apl, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> apl))</span>
<span id="cb5-13"><a href="scrna-seq.html#cb5-13" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">RunHarmony</span>(apl, <span class="st">&quot;Sample&quot;</span>)</span>
<span id="cb5-14"><a href="scrna-seq.html#cb5-14" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(apl, <span class="at">reduction =</span> <span class="st">&quot;harmony&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb5-15"><a href="scrna-seq.html#cb5-15" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(apl, <span class="at">resolution =</span> <span class="fl">0.8</span>)</span>
<span id="cb5-16"><a href="scrna-seq.html#cb5-16" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">RunTSNE</span>(apl, <span class="at">reduction =</span> <span class="st">&quot;harmony&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">perplexity =</span> <span class="dv">50</span>)</span>
<span id="cb5-17"><a href="scrna-seq.html#cb5-17" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(apl, <span class="at">reduction =</span> <span class="st">&quot;harmony&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb5-18"><a href="scrna-seq.html#cb5-18" tabindex="-1"></a></span>
<span id="cb5-19"><a href="scrna-seq.html#cb5-19" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(apl, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">pt.size =</span> <span class="dv">1</span>, </span>
<span id="cb5-20"><a href="scrna-seq.html#cb5-20" tabindex="-1"></a>             <span class="at">raster =</span> <span class="cn">FALSE</span>, <span class="at">cols =</span> color.lib) <span class="sc">+</span> <span class="fu">theme_few</span>()</span>
<span id="cb5-21"><a href="scrna-seq.html#cb5-21" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out.path, <span class="st">&quot;/3.ctrl.cluster.pdf&quot;</span>), p, <span class="at">width =</span> <span class="dv">9</span>, <span class="at">height =</span> <span class="dv">7</span>)</span>
<span id="cb5-22"><a href="scrna-seq.html#cb5-22" tabindex="-1"></a></span>
<span id="cb5-23"><a href="scrna-seq.html#cb5-23" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(apl, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">pt.size =</span> <span class="dv">1</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">label.size =</span> <span class="dv">10</span>,</span>
<span id="cb5-24"><a href="scrna-seq.html#cb5-24" tabindex="-1"></a>             <span class="at">raster =</span> <span class="cn">FALSE</span>, <span class="at">cols =</span> color.lib) <span class="sc">+</span> <span class="fu">theme_few</span>()</span>
<span id="cb5-25"><a href="scrna-seq.html#cb5-25" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">paste0</span>(out.path, <span class="st">&quot;/3.ctrl.cluster.label.pdf&quot;</span>), p, <span class="at">width =</span> <span class="dv">9</span>, <span class="at">height =</span> <span class="dv">7</span>)</span></code></pre></div>
</div>
</div>
<div id="integration" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Integration<a href="scrna-seq.html#integration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="integration-of-apls-and-normal-bmmcs" class="section level3 hasAnchor" number="2.5.1">
<h3><span class="header-section-number">2.5.1</span> Integration of APLs and normal BMMCs<a href="scrna-seq.html#integration-of-apls-and-normal-bmmcs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="scrna-seq.html#cb6-1" tabindex="-1"></a>exp.data <span class="ot">&lt;-</span> <span class="fu">cbind</span>(apl<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>counts, bmmc<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>counts)</span>
<span id="cb6-2"><a href="scrna-seq.html#cb6-2" tabindex="-1"></a>meta.data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(apl<span class="sc">@</span>meta.data[, <span class="fu">c</span>(<span class="st">&quot;orig.ident&quot;</span>,<span class="st">&quot;CellType.BlueprintEncodeData&quot;</span>,<span class="st">&quot;CellType.HumanPrimaryCellAtlasData&quot;</span>,<span class="st">&quot;CellType.ImmGenData&quot;</span>,<span class="st">&quot;CellType.MonacoImmuneData&quot;</span>,<span class="st">&quot;CellType.NovershternHematopoieticData&quot;</span>)],</span>
<span id="cb6-3"><a href="scrna-seq.html#cb6-3" tabindex="-1"></a>                   bmmc<span class="sc">@</span>meta.data[, <span class="fu">c</span>(<span class="st">&quot;orig.ident&quot;</span>,<span class="st">&quot;CellType.BlueprintEncodeData&quot;</span>,<span class="st">&quot;CellType.HumanPrimaryCellAtlasData&quot;</span>,<span class="st">&quot;CellType.ImmGenData&quot;</span>,<span class="st">&quot;CellType.MonacoImmuneData&quot;</span>,<span class="st">&quot;CellType.NovershternHematopoieticData&quot;</span>)])</span>
<span id="cb6-4"><a href="scrna-seq.html#cb6-4" tabindex="-1"></a></span>
<span id="cb6-5"><a href="scrna-seq.html#cb6-5" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> exp.data, <span class="at">project =</span> <span class="st">&quot;APL&quot;</span>, <span class="at">min.cells =</span> <span class="dv">0</span>, <span class="at">min.features =</span> <span class="dv">0</span>)</span>
<span id="cb6-6"><a href="scrna-seq.html#cb6-6" tabindex="-1"></a>apl[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(apl, <span class="at">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</span>
<span id="cb6-7"><a href="scrna-seq.html#cb6-7" tabindex="-1"></a>apl<span class="sc">@</span>meta.data <span class="ot">&lt;-</span> <span class="fu">cbind</span>(apl<span class="sc">@</span>meta.data, meta.data)</span></code></pre></div>
</div>
<div id="normalization" class="section level3 hasAnchor" number="2.5.2">
<h3><span class="header-section-number">2.5.2</span> Normalization<a href="scrna-seq.html#normalization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="scrna-seq.html#cb7-1" tabindex="-1"></a><span class="co"># The dimensionality reduction and clustering were performed using Seurat. </span></span>
<span id="cb7-2"><a href="scrna-seq.html#cb7-2" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(apl, <span class="at">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>, <span class="at">scale.factor =</span> <span class="fu">ncol</span>(apl) )</span>
<span id="cb7-3"><a href="scrna-seq.html#cb7-3" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(apl, <span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">3000</span>)</span>
<span id="cb7-4"><a href="scrna-seq.html#cb7-4" tabindex="-1"></a></span>
<span id="cb7-5"><a href="scrna-seq.html#cb7-5" tabindex="-1"></a><span class="co"># Identify the 10 most highly variable genes</span></span>
<span id="cb7-6"><a href="scrna-seq.html#cb7-6" tabindex="-1"></a>top10 <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">VariableFeatures</span>(apl), <span class="dv">10</span>)</span>
<span id="cb7-7"><a href="scrna-seq.html#cb7-7" tabindex="-1"></a></span>
<span id="cb7-8"><a href="scrna-seq.html#cb7-8" tabindex="-1"></a><span class="co"># Scale data</span></span>
<span id="cb7-9"><a href="scrna-seq.html#cb7-9" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(apl, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(apl))</span>
<span id="cb7-10"><a href="scrna-seq.html#cb7-10" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(apl, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> apl))</span></code></pre></div>
</div>
<div id="batch-effect-adjustment" class="section level3 hasAnchor" number="2.5.3">
<h3><span class="header-section-number">2.5.3</span> Batch effect adjustment<a href="scrna-seq.html#batch-effect-adjustment" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="scrna-seq.html#cb8-1" tabindex="-1"></a><span class="co"># Set batch</span></span>
<span id="cb8-2"><a href="scrna-seq.html#cb8-2" tabindex="-1"></a>apl<span class="sc">@</span>meta.data<span class="sc">$</span>Batch <span class="ot">&lt;-</span> <span class="fu">as.character</span>(apl<span class="sc">@</span>meta.data<span class="sc">$</span>Sample)</span>
<span id="cb8-3"><a href="scrna-seq.html#cb8-3" tabindex="-1"></a>apl<span class="sc">@</span>meta.data<span class="sc">$</span>Batch[<span class="fu">grep</span>(<span class="st">&quot;CTRL&quot;</span>, apl<span class="sc">$</span>Sample)] <span class="ot">=</span> <span class="st">&quot;CTRL&quot;</span></span>
<span id="cb8-4"><a href="scrna-seq.html#cb8-4" tabindex="-1"></a></span>
<span id="cb8-5"><a href="scrna-seq.html#cb8-5" tabindex="-1"></a><span class="co"># Batch effect adjustment</span></span>
<span id="cb8-6"><a href="scrna-seq.html#cb8-6" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">RunHarmony</span>(apl, <span class="st">&quot;Batch&quot;</span>, <span class="at">max.iter.harmony =</span> <span class="dv">2</span>, <span class="at">sigma =</span> <span class="fl">0.25</span>)</span>
<span id="cb8-7"><a href="scrna-seq.html#cb8-7" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(apl, <span class="at">reduction =</span> <span class="st">&quot;harmony&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb8-8"><a href="scrna-seq.html#cb8-8" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(apl, <span class="at">resolution =</span> <span class="dv">1</span>)</span>
<span id="cb8-9"><a href="scrna-seq.html#cb8-9" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(apl, <span class="at">reduction =</span> <span class="st">&quot;harmony&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span></code></pre></div>
</div>
<div id="cell-annotation" class="section level3 hasAnchor" number="2.5.4">
<h3><span class="header-section-number">2.5.4</span> Cell annotation<a href="scrna-seq.html#cell-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We use SingleR to provide cell type annotation.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="scrna-seq.html#cb9-1" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;SingleR/hs.BlueprintEncodeData.RDS&quot;</span>)</span>
<span id="cb9-2"><a href="scrna-seq.html#cb9-2" tabindex="-1"></a>pred.BlueprintEncodeData <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> apl<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data, <span class="at">ref =</span> ref, <span class="at">labels =</span> ref<span class="sc">$</span>label.main)</span>
<span id="cb9-3"><a href="scrna-seq.html#cb9-3" tabindex="-1"></a></span>
<span id="cb9-4"><a href="scrna-seq.html#cb9-4" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;SingleR/hs.HumanPrimaryCellAtlasData.RDS&quot;</span>)</span>
<span id="cb9-5"><a href="scrna-seq.html#cb9-5" tabindex="-1"></a>pred.HumanPrimaryCellAtlasData <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> apl<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data, <span class="at">ref =</span> ref, <span class="at">labels =</span> ref<span class="sc">$</span>label.main)</span>
<span id="cb9-6"><a href="scrna-seq.html#cb9-6" tabindex="-1"></a></span>
<span id="cb9-7"><a href="scrna-seq.html#cb9-7" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;SingleR/ImmGenData.RDS&quot;</span>)</span>
<span id="cb9-8"><a href="scrna-seq.html#cb9-8" tabindex="-1"></a>pred.ImmGenData <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> apl<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data, <span class="at">ref =</span> ref, <span class="at">labels =</span> ref<span class="sc">$</span>label.main)</span>
<span id="cb9-9"><a href="scrna-seq.html#cb9-9" tabindex="-1"></a></span>
<span id="cb9-10"><a href="scrna-seq.html#cb9-10" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;SingleR/MonacoImmuneData.RDS&quot;</span>)</span>
<span id="cb9-11"><a href="scrna-seq.html#cb9-11" tabindex="-1"></a>pred.MonacoImmuneData <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> apl<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data, <span class="at">ref =</span> ref, <span class="at">labels =</span> ref<span class="sc">$</span>label.main)</span>
<span id="cb9-12"><a href="scrna-seq.html#cb9-12" tabindex="-1"></a></span>
<span id="cb9-13"><a href="scrna-seq.html#cb9-13" tabindex="-1"></a>ref <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;SingleR/NovershternHematopoieticData.RDS&quot;</span>)</span>
<span id="cb9-14"><a href="scrna-seq.html#cb9-14" tabindex="-1"></a>pred.NovershternHematopoieticData <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> apl<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data, <span class="at">ref =</span> ref, <span class="at">labels =</span> ref<span class="sc">$</span>label.main)</span></code></pre></div>
</div>
</div>
<div id="apl-blasts" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> APL blasts<a href="scrna-seq.html#apl-blasts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="subset-apl-blasts" class="section level3 hasAnchor" number="2.6.1">
<h3><span class="header-section-number">2.6.1</span> Subset APL blasts<a href="scrna-seq.html#subset-apl-blasts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="scrna-seq.html#cb10-1" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">subset</span>(apl, Lineage <span class="sc">==</span> <span class="st">&quot;Leukemia_cell&quot;</span>)</span></code></pre></div>
</div>
<div id="normalization-1" class="section level3 hasAnchor" number="2.6.2">
<h3><span class="header-section-number">2.6.2</span> Normalization<a href="scrna-seq.html#normalization-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="scrna-seq.html#cb11-1" tabindex="-1"></a><span class="co"># The dimensionality reduction and clustering were performed using Seurat. </span></span>
<span id="cb11-2"><a href="scrna-seq.html#cb11-2" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(apl, <span class="at">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>, <span class="at">scale.factor =</span> <span class="fu">ncol</span>(apl) )</span>
<span id="cb11-3"><a href="scrna-seq.html#cb11-3" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(apl, <span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>, <span class="at">nfeatures =</span> <span class="dv">3000</span>)</span>
<span id="cb11-4"><a href="scrna-seq.html#cb11-4" tabindex="-1"></a></span>
<span id="cb11-5"><a href="scrna-seq.html#cb11-5" tabindex="-1"></a><span class="co"># Identify the 10 most highly variable genes</span></span>
<span id="cb11-6"><a href="scrna-seq.html#cb11-6" tabindex="-1"></a>top10 <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">VariableFeatures</span>(apl), <span class="dv">10</span>)</span>
<span id="cb11-7"><a href="scrna-seq.html#cb11-7" tabindex="-1"></a></span>
<span id="cb11-8"><a href="scrna-seq.html#cb11-8" tabindex="-1"></a><span class="co"># Scale data</span></span>
<span id="cb11-9"><a href="scrna-seq.html#cb11-9" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(apl, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(apl))</span>
<span id="cb11-10"><a href="scrna-seq.html#cb11-10" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(apl, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> apl))</span></code></pre></div>
</div>
<div id="batch-effect-adjustment-of-apl-blasts" class="section level3 hasAnchor" number="2.6.3">
<h3><span class="header-section-number">2.6.3</span> Batch effect adjustment of APL blasts<a href="scrna-seq.html#batch-effect-adjustment-of-apl-blasts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="scrna-seq.html#cb12-1" tabindex="-1"></a><span class="co"># Batch effect adjustment</span></span>
<span id="cb12-2"><a href="scrna-seq.html#cb12-2" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">RunHarmony</span>(apl, <span class="st">&quot;Sample&quot;</span>)</span>
<span id="cb12-3"><a href="scrna-seq.html#cb12-3" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(apl, <span class="at">reduction =</span> <span class="st">&quot;harmony&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb12-4"><a href="scrna-seq.html#cb12-4" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(apl, <span class="at">resolution =</span> <span class="dv">1</span>)</span>
<span id="cb12-5"><a href="scrna-seq.html#cb12-5" tabindex="-1"></a>apl <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(apl, <span class="at">reduction =</span> <span class="st">&quot;harmony&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span></code></pre></div>
</div>
</div>
<div id="viper" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> VIPER<a href="scrna-seq.html#viper" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="scrna-seq.html#cb13-1" tabindex="-1"></a>out.path.tf <span class="ot">&lt;-</span> <span class="fu">paste0</span>(out.path, <span class="st">&quot;/viper_APL_vs_CTRL_Granulocyte_1w_cell_per1000&quot;</span>)</span>
<span id="cb13-2"><a href="scrna-seq.html#cb13-2" tabindex="-1"></a>cmd <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;mkdir %s&quot;</span>, out.path.tf)</span>
<span id="cb13-3"><a href="scrna-seq.html#cb13-3" tabindex="-1"></a><span class="fu">system</span>(cmd)</span>
<span id="cb13-4"><a href="scrna-seq.html#cb13-4" tabindex="-1"></a></span>
<span id="cb13-5"><a href="scrna-seq.html#cb13-5" tabindex="-1"></a></span>
<span id="cb13-6"><a href="scrna-seq.html#cb13-6" tabindex="-1"></a><span class="fu">library</span>(viper)</span>
<span id="cb13-7"><a href="scrna-seq.html#cb13-7" tabindex="-1"></a></span>
<span id="cb13-8"><a href="scrna-seq.html#cb13-8" tabindex="-1"></a></span>
<span id="cb13-9"><a href="scrna-seq.html#cb13-9" tabindex="-1"></a>apl.sub <span class="ot">&lt;-</span> <span class="fu">subset</span>(apl, <span class="at">cells =</span> sub.name)</span>
<span id="cb13-10"><a href="scrna-seq.html#cb13-10" tabindex="-1"></a>apl.sub <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(apl.sub, <span class="at">features =</span> <span class="fu">rownames</span>(apl))</span>
<span id="cb13-11"><a href="scrna-seq.html#cb13-11" tabindex="-1"></a></span>
<span id="cb13-12"><a href="scrna-seq.html#cb13-12" tabindex="-1"></a></span>
<span id="cb13-13"><a href="scrna-seq.html#cb13-13" tabindex="-1"></a>pdata <span class="ot">&lt;-</span> apl.sub<span class="sc">@</span>meta.data[, <span class="fu">c</span>(<span class="st">&quot;orig.ident&quot;</span>,<span class="st">&quot;Lineage&quot;</span>,<span class="st">&quot;Sample&quot;</span>)]</span>
<span id="cb13-14"><a href="scrna-seq.html#cb13-14" tabindex="-1"></a><span class="fu">colnames</span>(pdata) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;Cell&#39;</span>, <span class="st">&#39;CellType&#39;</span>,<span class="st">&#39;Sample&#39;</span>)</span>
<span id="cb13-15"><a href="scrna-seq.html#cb13-15" tabindex="-1"></a>pdata<span class="sc">$</span>CellType <span class="ot">&lt;-</span> <span class="fu">as.character</span>(pdata<span class="sc">$</span>CellType)</span>
<span id="cb13-16"><a href="scrna-seq.html#cb13-16" tabindex="-1"></a>pdata<span class="sc">$</span>Cell <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pdata)</span>
<span id="cb13-17"><a href="scrna-seq.html#cb13-17" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">labelDescription=</span><span class="fu">c</span>(<span class="st">&#39;Cell&#39;</span>, <span class="st">&#39;CellType&#39;</span>,<span class="st">&#39;Sample&#39;</span>),</span>
<span id="cb13-18"><a href="scrna-seq.html#cb13-18" tabindex="-1"></a>                       <span class="at">row.names=</span><span class="fu">c</span>(<span class="st">&#39;Cell&#39;</span>, <span class="st">&#39;CellType&#39;</span>,<span class="st">&#39;Sample&#39;</span>))</span>
<span id="cb13-19"><a href="scrna-seq.html#cb13-19" tabindex="-1"></a>phenoData <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">&quot;AnnotatedDataFrame&quot;</span>, <span class="at">data =</span> pdata, <span class="at">varMetadata =</span> metadata)</span>
<span id="cb13-20"><a href="scrna-seq.html#cb13-20" tabindex="-1"></a></span>
<span id="cb13-21"><a href="scrna-seq.html#cb13-21" tabindex="-1"></a>exp.mat <span class="ot">&lt;-</span> apl.sub<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>scale.data[<span class="fu">rowSums</span>(apl.sub<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>scale.data <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">&lt;</span> <span class="fu">ncol</span>(apl.sub<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>scale.data)<span class="sc">*</span><span class="fl">0.1</span>, ]</span>
<span id="cb13-22"><a href="scrna-seq.html#cb13-22" tabindex="-1"></a>dset <span class="ot">&lt;-</span> <span class="fu">ExpressionSet</span>(<span class="at">assayData=</span>exp.mat,</span>
<span id="cb13-23"><a href="scrna-seq.html#cb13-23" tabindex="-1"></a>                      <span class="at">phenoData=</span>phenoData,</span>
<span id="cb13-24"><a href="scrna-seq.html#cb13-24" tabindex="-1"></a>                      <span class="at">annotation=</span><span class="st">&quot;RNAseq&quot;</span>)</span>
<span id="cb13-25"><a href="scrna-seq.html#cb13-25" tabindex="-1"></a></span>
<span id="cb13-26"><a href="scrna-seq.html#cb13-26" tabindex="-1"></a><span class="co">#sig &lt;- rowTtest(dset, &quot;CellType&quot;, c(&quot;LSC_like&quot;), &quot;HSPCs&quot;)$statistic</span></span>
<span id="cb13-27"><a href="scrna-seq.html#cb13-27" tabindex="-1"></a></span>
<span id="cb13-28"><a href="scrna-seq.html#cb13-28" tabindex="-1"></a><span class="fu">table</span>(pdata<span class="sc">$</span>CellType)</span>
<span id="cb13-29"><a href="scrna-seq.html#cb13-29" tabindex="-1"></a>signature <span class="ot">&lt;-</span> <span class="fu">rowTtest</span>(dset, <span class="st">&quot;CellType&quot;</span>, <span class="fu">c</span>(<span class="st">&quot;Leukemia_cell&quot;</span>), <span class="st">&quot;Granulocyte&quot;</span>)</span>
<span id="cb13-30"><a href="scrna-seq.html#cb13-30" tabindex="-1"></a>nullmodel <span class="ot">&lt;-</span> <span class="fu">ttestNull</span>(dset, <span class="st">&quot;CellType&quot;</span>, <span class="fu">c</span>(<span class="st">&quot;Leukemia_cell&quot;</span>), <span class="st">&quot;Granulocyte&quot;</span>, <span class="at">per =</span> <span class="dv">100</span>, <span class="at">repos =</span> <span class="cn">TRUE</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-31"><a href="scrna-seq.html#cb13-31" tabindex="-1"></a>signature <span class="ot">&lt;-</span> signature<span class="sc">$</span>statistic</span>
<span id="cb13-32"><a href="scrna-seq.html#cb13-32" tabindex="-1"></a></span>
<span id="cb13-33"><a href="scrna-seq.html#cb13-33" tabindex="-1"></a>mrs <span class="ot">&lt;-</span> <span class="fu">msviper</span>(signature, regulon, nullmodel)</span>
<span id="cb13-34"><a href="scrna-seq.html#cb13-34" tabindex="-1"></a><span class="fu">saveRDS</span>(mrs, <span class="fu">paste0</span>(out.path.tf, <span class="st">&quot;/viper.mra.rds&quot;</span>))</span>
<span id="cb13-35"><a href="scrna-seq.html#cb13-35" tabindex="-1"></a><span class="fu">saveRDS</span>(signature, <span class="fu">paste0</span>(out.path.tf, <span class="st">&quot;/viper.sig.rds&quot;</span>))</span>
<span id="cb13-36"><a href="scrna-seq.html#cb13-36" tabindex="-1"></a><span class="fu">saveRDS</span>(nullmodel, <span class="fu">paste0</span>(out.path.tf, <span class="st">&quot;/viper.dnull.rds&quot;</span>))</span></code></pre></div>
</div>
<div id="dorothea" class="section level2 hasAnchor" number="2.8">
<h2><span class="header-section-number">2.8</span> dorothea<a href="scrna-seq.html#dorothea" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="scrna-seq.html#cb14-1" tabindex="-1"></a><span class="fu">library</span>(dorothea)</span>
<span id="cb14-2"><a href="scrna-seq.html#cb14-2" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb14-3"><a href="scrna-seq.html#cb14-3" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb14-4"><a href="scrna-seq.html#cb14-4" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb14-5"><a href="scrna-seq.html#cb14-5" tabindex="-1"></a></span>
<span id="cb14-6"><a href="scrna-seq.html#cb14-6" tabindex="-1"></a></span>
<span id="cb14-7"><a href="scrna-seq.html#cb14-7" tabindex="-1"></a>out.path.tf <span class="ot">&lt;-</span> <span class="fu">paste0</span>(out.path, <span class="st">&quot;/dorothea_all_lineage_1w_cell&quot;</span>)</span>
<span id="cb14-8"><a href="scrna-seq.html#cb14-8" tabindex="-1"></a>cmd <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;mkdir %s&quot;</span>, out.path.tf)</span>
<span id="cb14-9"><a href="scrna-seq.html#cb14-9" tabindex="-1"></a><span class="fu">system</span>(cmd)</span>
<span id="cb14-10"><a href="scrna-seq.html#cb14-10" tabindex="-1"></a></span>
<span id="cb14-11"><a href="scrna-seq.html#cb14-11" tabindex="-1"></a>dorothea_regulon_human <span class="ot">&lt;-</span> <span class="fu">get</span>(<span class="fu">data</span>(<span class="st">&quot;dorothea_hs&quot;</span>, <span class="at">package =</span> <span class="st">&quot;dorothea&quot;</span>))</span>
<span id="cb14-12"><a href="scrna-seq.html#cb14-12" tabindex="-1"></a></span>
<span id="cb14-13"><a href="scrna-seq.html#cb14-13" tabindex="-1"></a>regulon <span class="ot">&lt;-</span> dorothea_regulon_human <span class="sc">%&gt;%</span></span>
<span id="cb14-14"><a href="scrna-seq.html#cb14-14" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(confidence <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;B&quot;</span>,<span class="st">&quot;C&quot;</span>,<span class="st">&quot;D&quot;</span>,<span class="st">&quot;E&quot;</span>))</span>
<span id="cb14-15"><a href="scrna-seq.html#cb14-15" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(regulon<span class="sc">$</span>tf))</span>
<span id="cb14-16"><a href="scrna-seq.html#cb14-16" tabindex="-1"></a></span>
<span id="cb14-17"><a href="scrna-seq.html#cb14-17" tabindex="-1"></a></span>
<span id="cb14-18"><a href="scrna-seq.html#cb14-18" tabindex="-1"></a>apl.sub <span class="ot">&lt;-</span> <span class="fu">run_viper</span>(apl.sub, regulon,</span>
<span id="cb14-19"><a href="scrna-seq.html#cb14-19" tabindex="-1"></a>                     <span class="at">options =</span> <span class="fu">list</span>(<span class="at">method =</span> <span class="st">&quot;scale&quot;</span>, <span class="at">minsize =</span> <span class="dv">4</span>, </span>
<span id="cb14-20"><a href="scrna-seq.html#cb14-20" tabindex="-1"></a>                                <span class="at">eset.filter =</span> <span class="cn">FALSE</span>, <span class="at">cores =</span> <span class="dv">1</span>, </span>
<span id="cb14-21"><a href="scrna-seq.html#cb14-21" tabindex="-1"></a>                                <span class="at">verbose =</span> <span class="cn">TRUE</span>))</span>
<span id="cb14-22"><a href="scrna-seq.html#cb14-22" tabindex="-1"></a></span>
<span id="cb14-23"><a href="scrna-seq.html#cb14-23" tabindex="-1"></a><span class="fu">DefaultAssay</span>(<span class="at">object =</span> apl.sub) <span class="ot">&lt;-</span> <span class="st">&quot;dorothea&quot;</span></span>
<span id="cb14-24"><a href="scrna-seq.html#cb14-24" tabindex="-1"></a>apl.sub <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(apl.sub)</span>
<span id="cb14-25"><a href="scrna-seq.html#cb14-25" tabindex="-1"></a></span>
<span id="cb14-26"><a href="scrna-seq.html#cb14-26" tabindex="-1"></a><span class="fu">dim</span>(apl.sub<span class="sc">@</span>assays<span class="sc">$</span>dorothea<span class="sc">@</span>scale.data)</span>
<span id="cb14-27"><a href="scrna-seq.html#cb14-27" tabindex="-1"></a></span>
<span id="cb14-28"><a href="scrna-seq.html#cb14-28" tabindex="-1"></a><span class="fu">dim</span>(regulon)</span>
<span id="cb14-29"><a href="scrna-seq.html#cb14-29" tabindex="-1"></a><span class="fu">sum</span>(regulon<span class="sc">$</span>tf <span class="sc">%in%</span> <span class="fu">rownames</span>(apl.sub))</span>
<span id="cb14-30"><a href="scrna-seq.html#cb14-30" tabindex="-1"></a><span class="fu">sum</span>(regulon<span class="sc">$</span>target <span class="sc">%in%</span> <span class="fu">rownames</span>(apl.sub))</span>
<span id="cb14-31"><a href="scrna-seq.html#cb14-31" tabindex="-1"></a></span>
<span id="cb14-32"><a href="scrna-seq.html#cb14-32" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(regulon<span class="sc">$</span>tf))</span>
<span id="cb14-33"><a href="scrna-seq.html#cb14-33" tabindex="-1"></a></span>
<span id="cb14-34"><a href="scrna-seq.html#cb14-34" tabindex="-1"></a></span>
<span id="cb14-35"><a href="scrna-seq.html#cb14-35" tabindex="-1"></a>dorothea_scores_df <span class="ot">&lt;-</span> <span class="fu">GetAssayData</span>(apl.sub, <span class="at">slot =</span> <span class="st">&quot;scale.data&quot;</span>, <span class="at">assay =</span> <span class="st">&quot;dorothea&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-36"><a href="scrna-seq.html#cb14-36" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">check.names =</span> F) <span class="sc">%&gt;%</span> <span class="fu">t</span>()</span>
<span id="cb14-37"><a href="scrna-seq.html#cb14-37" tabindex="-1"></a><span class="fu">saveRDS</span>(dorothea_scores_df, <span class="fu">paste0</span>(out.path.tf, <span class="st">&quot;/dorothea_scores_df.rds&quot;</span>))</span></code></pre></div>
</div>
<div id="cell-signature" class="section level2 hasAnchor" number="2.9">
<h2><span class="header-section-number">2.9</span> Cell signature<a href="scrna-seq.html#cell-signature" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The cell signature matrix for each branch in APL was generated using CIBERSORTx (<a href="https://cibersortx.stanford.edu/runcibersortx.php" class="uri">https://cibersortx.stanford.edu/runcibersortx.php</a>)</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="scrna-seq.html#cb15-1" tabindex="-1"></a>apl.signature <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/CIBERSORTx_sig_1w_branch_mix_out.txt&quot;</span>, <span class="at">header =</span> T, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb15-2"><a href="scrna-seq.html#cb15-2" tabindex="-1"></a>apl.signature <span class="sc">%&gt;%</span> <span class="fu">dim</span>()</span></code></pre></div>
<pre><code>## [1] 3388    9</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="scrna-seq.html#cb17-1" tabindex="-1"></a>apl.signature <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span>
<span id="cb17-2"><a href="scrna-seq.html#cb17-2" tabindex="-1"></a></span>
<span id="cb17-3"><a href="scrna-seq.html#cb17-3" tabindex="-1"></a><span class="fu">pheatmap</span>(<span class="fu">as.matrix</span>(apl.signature), <span class="at">scale =</span> <span class="st">&quot;row&quot;</span>,</span>
<span id="cb17-4"><a href="scrna-seq.html#cb17-4" tabindex="-1"></a>         <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">&quot;#1B3361&quot;</span>,<span class="st">&quot;#76C7FF&quot;</span>,<span class="st">&quot;#FFFFFF&quot;</span>,<span class="st">&quot;#FF987A&quot;</span>,<span class="st">&quot;#6A0D28&quot;</span>))(<span class="dv">501</span>),</span>
<span id="cb17-5"><a href="scrna-seq.html#cb17-5" tabindex="-1"></a>         <span class="at">cluster_row =</span> T, <span class="at">cluster_col =</span> T, <span class="at">border_color =</span> <span class="cn">NA</span>,</span>
<span id="cb17-6"><a href="scrna-seq.html#cb17-6" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">&quot;manhattan&quot;</span>,</span>
<span id="cb17-7"><a href="scrna-seq.html#cb17-7" tabindex="-1"></a>         <span class="at">clustering_distance_cols =</span> <span class="st">&quot;manhattan&quot;</span>,</span>
<span id="cb17-8"><a href="scrna-seq.html#cb17-8" tabindex="-1"></a>         <span class="at">clustering_method =</span> <span class="st">&quot;ward.D&quot;</span>,</span>
<span id="cb17-9"><a href="scrna-seq.html#cb17-9" tabindex="-1"></a>         <span class="at">fontsize_col =</span> <span class="dv">10</span>,</span>
<span id="cb17-10"><a href="scrna-seq.html#cb17-10" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="fl">0.1</span>)</span></code></pre></div>
<p><img src="APL-stemness_files/figure-html/unnamed-chunk-15-1.png" width="480" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="bulk.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/01-scRNAseq.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["APL-stemness.pdf", "APL-stemness.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
